<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Tae Kim" />


<title>Analysis of African American Co-expression Network</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Differential Coexpression Network</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="AA_EA.html">Discrete Example</a>
</li>
<li>
  <a href="AA.html">Continuous Example</a>
</li>
<li>
  <a href="W_and_V_alt.html">Simulations</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tk382/networkAnc">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis of African American Co-expression Network</h1>
<h4 class="author">Tae Kim</h4>
<h4 class="date">7/9/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>diffNet/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190331code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190331)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190331code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190331)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtk382networkAnctree906e3002fbb321e226aca4d9ca1e17efc73c4fedtargetblank906e300a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tk382/networkAnc/tree/906e3002fbb321e226aca4d9ca1e17efc73c4fed" target="_blank">906e300</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtk382networkAnctree906e3002fbb321e226aca4d9ca1e17efc73c4fedtargetblank906e300a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/.DS_Store
    Ignored:    R/.Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/AA_EA_cache/
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store
    Ignored:    src/.DS_Store

Untracked files:
    Untracked:  .Rprofile
    Untracked:  .gitignore
    Untracked:  analysis/figure/AA_EA.Rmd/
    Untracked:  data/
    Untracked:  docs/figure/liquid_assoc.Rmd/
    Untracked:  docs/figure/simulation1.Rmd/
    Untracked:  docs/figure/simulation2.Rmd/
    Untracked:  docs/figure/simulation3.Rmd/
    Untracked:  docs/figure/simulation4.Rmd/
    Untracked:  man/do_polyroot_arma.Rd
    Untracked:  man/do_polyroot_cpp.Rd
    Untracked:  man/post_score.Rd
    Untracked:  old_analysis/
    Untracked:  targeted_TF_AA.txt
    Untracked:  wandv.R

Unstaged changes:
    Modified:   .Rbuildignore
    Modified:   DESCRIPTION
    Modified:   NAMESPACE
    Deleted:    analysis/AA_vs_EA_skin_not_sun_exposed.Rmd
    Deleted:    analysis/AA_vs_EA_skin_not_sun_exposed.pdf
    Modified:   analysis/W_and_V_alt.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    analysis/score_test.R
    Modified:   docs/.nojekyll
    Modified:   dynamic_coexpression_githubversion.tex
    Deleted:    src/diffNet.so
    Modified:   src/utilities.cpp

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/906e3002fbb321e226aca4d9ca1e17efc73c4fed/analysis/AA.Rmd" target="_blank">906e300</a>
</td>
<td>
tk382
</td>
<td>
2019-07-11
</td>
<td>
wflow_publish(c(“analysis/AA.Rmd”, “analysis/AA_EA.Rmd”), republish = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/acc3d145a84398e64d7e0a6fe5d3bf1c6cd376e0/analysis/AA.Rmd" target="_blank">acc3d14</a>
</td>
<td>
tk382
</td>
<td>
2019-07-04
</td>
<td>
modified analysis files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/68a6d1e9fb9b48920d7f61f57d34cab888cd7057/docs/AA.html" target="_blank">68a6d1e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/AA.html" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/dc21566d1401835d1d813aceac5a26cd6e12d8e8/docs/AA.html" target="_blank">dc21566</a>
</td>
<td>
tk382
</td>
<td>
2019-04-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/368bd3ed13c957720160a6db953a0de64a05ac07/docs/AA.html" target="_blank">368bd3e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/00c00595f30810d0481eb5f9e19924bbdb5248ac/analysis/AA.Rmd" target="_blank">00c0059</a>
</td>
<td>
tk382
</td>
<td>
2019-04-02
</td>
<td>
wflow_publish(c(“analysis/AA.Rmd”, “analysis/AA_EA.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/4053f506a8258d01fa5c3025658cb85e0e6651d2/docs/AA.html" target="_blank">4053f50</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f68ec1044ebda6ff8908b6ca6554093dc1a17d0/analysis/AA.Rmd" target="_blank">1f68ec1</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
wflow_publish(“analysis/AA.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/AA.html" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/058b38f74cb2d6e2e8ba245d117be04a2d7bf035/analysis/AA.Rmd" target="_blank">058b38f</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/license.Rmd”, “analysis/about.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-pre-processing" class="section level1">
<h1>1. Data pre-processing</h1>
<p>We use the RNA-seq expression level data provided by GTEx, which has been already pre-processed. We choose to study not-sun-exposed suprapubic skin tissue as a pilot study because we expect to see some ancestry effect on the expression level on the skin tissue. Genes were selected based on expression threshold of <span class="math inline">\(&gt;0.1\)</span> RPKM in at least 10 individuals and <span class="math inline">\(\geq\)</span> reads in at least 10 individuals. Then the expression values were quantile normalized to the average empirical distribution observed across samples. That means, each gene has the same distribution across the individuals. Then they were inverse quantile normalized to fit standard normal distribution.</p>
<div id="a-load-and-clean-the-data" class="section level2">
<h2>(a) Load and clean the data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;data/skin_not_sun_exposed_suprapubic.Rdata&quot;</span>)
exp =<span class="st"> </span>skin; <span class="kw">rm</span>(skin)
<span class="kw">colnames</span>(exp) =<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;[.]&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">colnames</span>(exp))
ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colSums</span>(<span class="op">!</span><span class="kw">is.na</span>(exp)) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)
exp =<span class="st"> </span>exp[,<span class="op">-</span>ind]
ind=<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowSums</span>(<span class="op">!</span><span class="kw">is.na</span>(exp)) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)
exp =<span class="st"> </span>exp[<span class="op">-</span>ind, ]
<span class="kw">invisible</span>(<span class="kw">gc</span>())
finalA =<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/finalA.txt&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>)
AAs =<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">colnames</span>(exp), finalA<span class="op">$</span>subject)
EAs =<span class="st"> </span><span class="kw">colnames</span>(exp)[<span class="op">!</span><span class="st"> </span><span class="kw">colnames</span>(exp) <span class="op">%in%</span><span class="st"> </span>finalA<span class="op">$</span>subject]
newA=<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">subject =</span> AAs, <span class="dt">A =</span> finalA[<span class="kw">match</span>(AAs, finalA<span class="op">$</span>subject), <span class="dv">2</span>])
Y =<span class="st"> </span>exp[, <span class="kw">c</span>(AAs, EAs)]
<span class="kw">rm</span>(exp); <span class="kw">invisible</span>(<span class="kw">gc</span>())
EA_A =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">subject =</span> EAs, <span class="dt">A =</span> <span class="dv">0</span>)
X =<span class="st"> </span><span class="kw">rbind</span>(newA, EA_A)
<span class="kw">rm</span>(newA, EA_A, AAs, EAs, ind, finalA)
<span class="kw">rownames</span>(Y) =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(Y), <span class="st">&#39;[.]&#39;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])
Y =<span class="st"> </span><span class="kw">t</span>(Y)
ind =<span class="st"> </span><span class="kw">which</span>(X<span class="op">$</span>A <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)
Y =<span class="st"> </span>Y[ind,]
X =<span class="st"> </span>X[ind, ]
orig =<span class="st"> </span>Y</code></pre></div>
</div>
<div id="b-scale-the-data-and-regress-out-the-mean" class="section level2">
<h2>(b) scale the data and regress out the mean</h2>
<p>Wefirst scale the gene expression level matrix to have variance 1. Then we regress out ancestry from the gene expression level to remove the mean effect.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">orig_A =<span class="st"> </span>X<span class="op">$</span>A
X<span class="op">$</span>A =<span class="st"> </span><span class="kw">scale</span>(X<span class="op">$</span>A) <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)<span class="op">-</span><span class="dv">1</span>)
Y =<span class="st"> </span><span class="kw">resid</span>(<span class="kw">lm</span>(Y<span class="op">~</span>X<span class="op">$</span>A))
Y =<span class="st"> </span><span class="kw">scale</span>(Y) <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)<span class="op">-</span><span class="dv">1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#reference chi squared distribution to compare the null distribution</span>
ref_chisq =<span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># number of permutations</span>
B =<span class="st"> </span><span class="dv">5000</span></code></pre></div>
</div>
<div id="c-match-ensembl-id-and-hgnc-symbols" class="section level2">
<h2>(c) match ENSEMBL ID and hgnc symbols</h2>
<p>Using data set from <a href="https://www.genenames.org/download/custom/" class="uri">https://www.genenames.org/download/custom/</a>, we match the Ensembl gene IDs to HGNC symbols for better readability.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hgnc_names =<span class="st"> </span>hgnc[<span class="kw">match</span>(<span class="kw">colnames</span>(Y), hgnc<span class="op">$</span><span class="st">`</span><span class="dt">Ensembl gene ID</span><span class="st">`</span>), <span class="st">&quot;Approved symbol&quot;</span>]
ref =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">ensembl_gene_id =</span> <span class="kw">colnames</span>(Y), <span class="dt">hgnc_symbol =</span> hgnc_names)
ref<span class="op">$</span>ensembl_gene_id =<span class="st"> </span><span class="kw">as.character</span>(ref<span class="op">$</span>ensembl_gene_id)
ref<span class="op">$</span>hgnc_symbol =<span class="st"> </span><span class="kw">as.character</span>(ref<span class="op">$</span>hgnc_symbol)</code></pre></div>
</div>
</div>
<div id="transcription-factors-against-all-other-genes" class="section level1">
<h1>2. Transcription Factors against all other genes</h1>
<p>Since transcription factors usually regulate the transcription of multiple genes, we believe they were appropriate targets to study their variance in activity across different ancestry. We use the list of transcription factors from <a href="http://www.tfcheckpoint.org/" class="uri">http://www.tfcheckpoint.org/</a>.</p>
<div id="a-read-in-tf-data" class="section level2">
<h2>(a) Read in TF data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">TF =<span class="st"> </span><span class="kw">fread</span>(<span class="st">&quot;data/not_build/TFcheckpoint.txt&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>)
TF =<span class="st"> </span><span class="kw">setDF</span>(TF)
TF =<span class="st"> </span>TF[TF<span class="op">$</span>TFClass_human <span class="op">==</span><span class="st"> &quot;TFclass&quot;</span>, ]
TF =<span class="st"> </span>TF[,<span class="dv">1</span>]
TF_results =<span class="st"> </span>ref[<span class="kw">match</span>(TF, ref<span class="op">$</span>hgnc), ]
genes =<span class="st"> </span><span class="kw">colnames</span>(Y)[<span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span>TF_results<span class="op">$</span>ensembl_gene_id)]
numTFs =<span class="st"> </span><span class="kw">length</span>(genes)
TF_results =<span class="st"> </span>TF_results[<span class="kw">match</span>(genes, TF_results<span class="op">$</span>ensembl_gene_id), ]</code></pre></div>
</div>
<div id="b-test-all-tfs-against-the-rest-of-the-genes" class="section level2">
<h2>(b) Test all TFs against the rest of the genes</h2>
<p>For each transcription factor, we calculate its score statistics against all other genes. Then we add up the scores for each transcription factor and see if any stands out.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scores =<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="kw">length</span>(genes), <span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genes)){
  ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">==</span><span class="st"> </span>TF_results<span class="op">$</span>ensembl_gene_id[i])
  W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,ind], Y[, <span class="op">-</span>ind])
  V =<span class="st"> </span><span class="kw">store_V_c</span>(Y[,ind], Y[, <span class="op">-</span>ind])
  scores[i,] =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W, V)
}</code></pre></div>
</div>
<div id="c-plot-the-result" class="section level2">
<h2>(c) plot the result</h2>
<p>The plot below shows that some transcription factors stand out from the rest with high score statistics. The red line shows the expected value based on <span class="math inline">\(\chi^2\)</span> statistics. The right plot shows the qq-plot of score statistics of the strongest signal “FOXD1” against the expected <span class="math inline">\(\chi_1^2\)</span> distribution. Note that all the score statistics are highly correlated to one another, so that the calibration is poor.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
d =<span class="st"> </span><span class="kw">rowSums</span>(scores)
df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">d =</span> d, <span class="dt">genes =</span> TF_results<span class="op">$</span>hgnc_symbol, <span class="dt">xaxis =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(d))
<span class="kw">plot</span>(d, 
     <span class="dt">main =</span> <span class="st">&#39;Transcription Factors&#39;</span>,
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">length</span>(d)<span class="op">+</span><span class="dv">100</span>),
     <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(d) <span class="op">+</span><span class="st"> </span><span class="dv">10000</span>),
     <span class="dt">xlab =</span> <span class="st">&quot;&quot;</span>,
     <span class="dt">ylab =</span> <span class="st">&quot;d test statistic&quot;</span>)
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">25190</span>, <span class="dt">col=</span><span class="st">&#39;red&#39;</span>)
ind =<span class="st"> </span><span class="kw">which</span>(d <span class="op">&gt;</span><span class="st"> </span><span class="kw">ncol</span>(Y) <span class="op">+</span><span class="st"> </span><span class="dv">180</span> <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="dv">2</span><span class="op">*</span><span class="kw">ncol</span>(Y)))
df =<span class="st"> </span>df[ind, ]
<span class="kw">with</span>(df, <span class="kw">text</span>(d <span class="op">~</span><span class="st"> </span>xaxis, <span class="dt">labels =</span> genes, <span class="dt">pos =</span> <span class="dv">3</span>))

ref_chisq =<span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genes)){
    <span class="cf">if</span>(TF_results<span class="op">$</span>hgnc_symbol[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;FOXD1&quot;</span>)){
      <span class="kw">qqplot</span>(ref_chisq, scores[i, ], 
            <span class="dt">main =</span> <span class="kw">paste</span>(TF_results<span class="op">$</span>hgnc_symbol[i], <span class="st">&#39;unadjusted&#39;</span>),
            <span class="dt">cex =</span> <span class="fl">0.5</span>,
            <span class="dt">ylab =</span> <span class="st">&quot;observed distribution of score&quot;</span>,
            <span class="dt">xlab =</span> <span class="st">&quot;expected distribution&quot;</span>)
      <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
  }
}</code></pre></div>
<p><img src="figure/AA.Rmd/plot_allTFs-1.png" width="700" style="display: block; margin: auto;" /></p>

</div>
</div>
<div id="investigating-the-top-signals" class="section level1">
<h1>3. Investigating the top signals</h1>
<p>We take the top 3 transcription factors and perform permutation test to make inference about their degree statistics.</p>
<div id="a-foxd1-permutation-test-top-signals" class="section level2">
<h2>(a) FOXD1 permutation test &amp; top signals</h2>
<p>In order to verify if FOXD1 truly has a signal, we perform permutation test. Note that the highest test statistic does not mean the lowest p-value. Since the null distribution of high-dimensional case is not defined in closed form, we need to perform the permutation test to make inference. The p-value is indeed very small. We also look at top 2 target genes whose co-expression with FOXD1 varies strongly with respect to ancestry. We plot the product of the expression levels of the pair, which is intended to show the estimate of correlation at a given ancestry level. For these selected gene pairs, they seem to be highly co-expressed for Africans and lowly co-expressed for Europeans.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">foxd1 =<span class="st"> </span><span class="kw">make_permtest</span>(<span class="st">&quot;FOXD1&quot;</span>, <span class="dt">Y =</span> Y, <span class="dt">A =</span> X<span class="op">$</span>A)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">hist</span>(foxd1<span class="op">$</span>bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">10000</span>, <span class="kw">max</span>(<span class="kw">max</span>(foxd1<span class="op">$</span>bootstrapped_d), d))<span class="op">+</span><span class="dv">20</span>,
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;p value=&quot;</span>, foxd1<span class="op">$</span>p), <span class="dt">xlab =</span> <span class="st">&quot;Null distribution of d&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>foxd1<span class="op">$</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</code></pre></div>
<p><img src="figure/AA.Rmd/foxd_perm-1.png" width="700" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">diagnosis_plot</span>(foxd1, <span class="st">&quot;FOXD1&quot;</span>, <span class="dt">orig =</span> orig, <span class="dt">Y =</span> Y, <span class="dt">A =</span> orig_A)</code></pre></div>
<p><img src="figure/AA.Rmd/foxd_perm-2.png" width="700" style="display: block; margin: auto;" /></p>
</div>
<div id="b-ncor2-permutation-test-and-top-signals" class="section level2">
<h2>(b) NCOR2 permutation test and top signals</h2>
<p>We repeat the same analysis for the second strongest signal NCOR2.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ncor =<span class="st"> </span><span class="kw">make_permtest</span>(<span class="st">&quot;NCOR2&quot;</span>, <span class="dt">Y =</span> Y, <span class="dt">A =</span> X<span class="op">$</span>A)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">hist</span>(ncor<span class="op">$</span>bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">10000</span>, <span class="kw">max</span>(<span class="kw">max</span>(ncor<span class="op">$</span>bootstrapped_d), ncor<span class="op">$</span>d))<span class="op">+</span><span class="dv">20</span>,
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;p value=&quot;</span>, ncor<span class="op">$</span>p), <span class="dt">xlab =</span> <span class="st">&quot;Null distribution of d&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v =</span> ncor<span class="op">$</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</code></pre></div>
<p><img src="figure/AA.Rmd/ncor_perm_first-1.png" width="700" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">diagnosis_plot</span>(ncor, <span class="st">&quot;NCOR2&quot;</span>, <span class="dt">orig =</span> orig, <span class="dt">Y =</span> Y, <span class="dt">A =</span> orig_A)</code></pre></div>
<p><img src="figure/AA.Rmd/ncor_perm_first-2.png" width="700" style="display: block; margin: auto;" /></p>
</div>
<div id="c-zeb2-permutation-test-and-top-signals" class="section level2">
<h2>(c) ZEB2 permutation test and top signals</h2>
<p>We repeat the same analysis for the second strongest signal ZEB2.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ncor =<span class="st"> </span><span class="kw">make_permtest</span>(<span class="st">&quot;ZEB2&quot;</span>, <span class="dt">Y =</span> Y, <span class="dt">A =</span> X<span class="op">$</span>A)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">hist</span>(ncor<span class="op">$</span>bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">10000</span>, <span class="kw">max</span>(<span class="kw">max</span>(ncor<span class="op">$</span>bootstrapped_d), ncor<span class="op">$</span>d))<span class="op">+</span><span class="dv">20</span>,
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;p value=&quot;</span>, ncor<span class="op">$</span>p), <span class="dt">xlab =</span> <span class="st">&quot;Null distribution of d&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v =</span> ncor<span class="op">$</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</code></pre></div>
<p><img src="figure/AA.Rmd/ncor_perm-1.png" width="700" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">diagnosis_plot</span>(ncor, <span class="st">&quot;ZEB2&quot;</span>, <span class="dt">orig =</span> orig, <span class="dt">Y =</span> Y, <span class="dt">A =</span> orig_A)</code></pre></div>
<p><img src="figure/AA.Rmd/ncor_perm-2.png" width="700" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="remove-outliers-samples-with-ancestry-less-than-0.6" class="section level1">
<h1>4. Remove outliers (samples with ancestry less than 0.6)</h1>
<p>We suspect that the outliers with low global ancestry could be driving most of the signals. So we repeat the same analysis after we remove the samples with global ancestry less than 0.6.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(orig_A, <span class="dt">ylab =</span> <span class="st">&#39;global ancestry&#39;</span>, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>, <span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>)
ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.6</span>)
<span class="kw">points</span>(ind, orig_A[ind], <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA.Rmd/remove_outliers-1.png" width="700" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">remove =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.6</span>)
newY =<span class="st"> </span>orig[<span class="op">-</span>remove,]
newX =<span class="st"> </span>X[<span class="op">-</span>remove,]
newX<span class="op">$</span>A =<span class="st"> </span><span class="kw">scale</span>(newX<span class="op">$</span>A) <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(newX<span class="op">$</span>A)) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(newX<span class="op">$</span>A)<span class="op">-</span><span class="dv">1</span>)
newY =<span class="st"> </span><span class="kw">resid</span>(<span class="kw">lm</span>(newY<span class="op">~</span>newX<span class="op">$</span>A))
newY =<span class="st"> </span><span class="kw">scale</span>(newY) <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(newX<span class="op">$</span>A)) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(newX<span class="op">$</span>A)<span class="op">-</span><span class="dv">1</span>)</code></pre></div>
<div id="a-re-run-the-transcription-factors" class="section level2">
<h2>(a) Re-run the transcription factors</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scores =<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="kw">length</span>(genes), <span class="kw">ncol</span>(newY)<span class="op">-</span><span class="dv">1</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genes)){
  ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(newY) <span class="op">==</span><span class="st"> </span>TF_results<span class="op">$</span>ensembl_gene_id[i])
  W =<span class="st"> </span><span class="kw">store_W_c</span>(newY[,ind], newY[, <span class="op">-</span>ind])
  V =<span class="st"> </span><span class="kw">store_V_c</span>(newY[,ind], newY[, <span class="op">-</span>ind])
  scores[i,] =<span class="st"> </span><span class="kw">get_score_W_c</span>(newX<span class="op">$</span>A, W, V)
}

<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
d =<span class="st"> </span><span class="kw">rowSums</span>(scores)
df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">d =</span> d, <span class="dt">genes =</span> TF_results<span class="op">$</span>hgnc_symbol, <span class="dt">xaxis =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(d))
<span class="kw">plot</span>(d, 
     <span class="dt">ylab =</span> <span class="st">&#39;d&#39;</span>, 
     <span class="dt">main =</span> <span class="st">&#39;Transcription Factors&#39;</span>,
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">length</span>(d)<span class="op">+</span><span class="dv">100</span>),
     <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(d) <span class="op">+</span><span class="st"> </span><span class="dv">10000</span>),
     <span class="dt">cex =</span> <span class="fl">0.5</span>)
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">25190</span>, <span class="dt">col=</span><span class="st">&#39;red&#39;</span>)
ind =<span class="st"> </span><span class="kw">which</span>(d <span class="op">&gt;</span><span class="st"> </span><span class="kw">ncol</span>(newY) <span class="op">+</span><span class="st"> </span><span class="dv">200</span> <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="dv">2</span><span class="op">*</span><span class="kw">ncol</span>(newY)))
<span class="cf">if</span>(<span class="kw">length</span>(ind)<span class="op">&gt;</span><span class="dv">0</span>){
  df =<span class="st"> </span>df[ind, ]
  <span class="kw">with</span>(df, <span class="kw">text</span>(d <span class="op">~</span><span class="st"> </span>xaxis, <span class="dt">labels =</span> genes, <span class="dt">pos =</span> <span class="dv">3</span>))
}

ref_chisq =<span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(newY)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span>)
ind =<span class="st"> </span><span class="kw">which.max</span>(d)
<span class="kw">qqplot</span>(ref_chisq, scores[ind,],
       <span class="dt">main =</span> TF_results<span class="op">$</span>hgnc_symbol[ind],
       <span class="dt">cex =</span> <span class="fl">0.5</span>)
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA.Rmd/tf_no_outliers-1.png" width="700" style="display: block; margin: auto;" /></p>
</div>
<div id="b-top-signal-twist1" class="section level2">
<h2>(b) Top signal: TWIST1</h2>
<div id="i-permutation-test" class="section level3">
<h3>(i) Permutation Test</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">twist =<span class="st"> </span><span class="kw">make_permtest</span>(<span class="st">&quot;TWIST1&quot;</span>, <span class="dt">Y =</span> newY, <span class="dt">A =</span> newX<span class="op">$</span>A)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">hist</span>(twist<span class="op">$</span>bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="kw">min</span>(twist<span class="op">$</span>bootstrapped_d)<span class="op">-</span><span class="dv">100</span>, <span class="kw">max</span>(<span class="kw">max</span>(twist<span class="op">$</span>bootstrapped_d), twist<span class="op">$</span>d)),
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;p-value = &quot;</span>, twist<span class="op">$</span>p))
<span class="kw">abline</span>(<span class="dt">v =</span> twist<span class="op">$</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</code></pre></div>
<p><img src="figure/AA.Rmd/twist_perm-1.png" width="700" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">diagnosis_plot</span>(twist, <span class="st">&quot;TWIST1&quot;</span>, <span class="dt">orig =</span> orig[<span class="op">-</span>remove, ], <span class="dt">Y =</span> newY, <span class="dt">A =</span> newX<span class="op">$</span>A)</code></pre></div>
<p><img src="figure/AA.Rmd/twist_perm-2.png" width="700" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="targeted-analysis-of-transcription-factors" class="section level1">
<h1>5. Targeted analysis of transcription factors</h1>
<div id="a-test-all-tfs-against-target-genes" class="section level2">
<h2>(a) Test all TFs against target genes</h2>
<p>The database of TF2DNA that shows the target genes for each transcription factor. We study the 1306 transcription factors introduced in this data set. Instead of measuring the co-expression against all the other genes, we only look at teh score statistics against the target genes with top binding scores.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(data.table)
TFs =<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/&quot;</span>)
TFs =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(TFs, <span class="st">&quot;.pscan&quot;</span>), <span class="cf">function</span>(x) x[[<span class="dv">1</span>]])
results =<span class="st"> </span>ref[<span class="kw">match</span>(TFs, ref<span class="op">$</span>hgnc_symbol), ]
results =<span class="st"> </span>results[<span class="op">!</span><span class="kw">is.na</span>(results<span class="op">$</span>hgnc_symbol), ]
out =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">TF =</span> results<span class="op">$</span>hgnc_symbol, 
                 <span class="dt">numTargets =</span> <span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(results)),
                 <span class="dt">degree =</span> <span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(results)))</code></pre></div>
<p>For each transcription factor, we need to read in the target data, which can take very long due to a high number of file IO.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(results<span class="op">$</span>hgnc_symbol)){
  targets =<span class="st"> </span><span class="kw">fread</span>(<span class="kw">paste0</span>(<span class="st">&quot;/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/&quot;</span>, results<span class="op">$</span>hgnc_symbol[i], <span class="st">&quot;.pscan&quot;</span>))
  targets =<span class="st"> </span>targets[<span class="kw">which</span>(targets<span class="op">$</span>p_value <span class="op">==</span><span class="st"> </span><span class="kw">min</span>(targets<span class="op">$</span>p_value)), <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">6</span>,<span class="dv">7</span>)]
  target_genenames =<span class="st"> </span>ref[<span class="kw">match</span>(targets<span class="op">$</span>target_name, ref<span class="op">$</span>hgnc_symbol), ]
  main_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">==</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id[i])
  target_ind =<span class="st"> </span><span class="kw">match</span>(target_genenames<span class="op">$</span>ensembl_gene_id, <span class="kw">colnames</span>(Y))
  target_ind =<span class="st"> </span>target_ind[<span class="op">!</span><span class="kw">is.na</span>(target_ind)]
  target_ind =<span class="st"> </span>target_ind[target_ind <span class="op">!=</span><span class="st"> </span>main_ind]
  W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,main_ind], Y[, target_ind])
  V =<span class="st"> </span><span class="kw">store_V_c</span>(Y[,main_ind], Y[, target_ind])
  scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W, V)
  out<span class="op">$</span>numTargets[i] =<span class="st"> </span><span class="kw">ncol</span>(W)
  out<span class="op">$</span>degree[i] =<span class="st"> </span><span class="kw">sum</span>(scores)
}</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d =<span class="st"> </span>out<span class="op">$</span>degree <span class="op">/</span><span class="st"> </span>out<span class="op">$</span>numTargets
df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">d =</span> d, 
                <span class="dt">genes =</span> results<span class="op">$</span>hgnc_symbol, 
                <span class="dt">xaxis =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(d))
<span class="kw">plot</span>(d,
     <span class="dt">main =</span> <span class="st">&#39;Targeted TF&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;averaged scores&#39;</span>,
     <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(d) <span class="op">+</span><span class="st"> </span><span class="fl">0.7</span>),
     <span class="dt">xlab =</span> <span class="kw">length</span>(d) <span class="op">+</span><span class="st"> </span><span class="dv">25</span>)
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">1</span>, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
ind =<span class="st"> </span><span class="kw">which</span>(d <span class="op">&gt;</span><span class="st"> </span><span class="fl">2.8</span>)
df =<span class="st"> </span>df[ind,]
<span class="kw">with</span>(df, <span class="kw">text</span>(d<span class="op">~</span>xaxis, <span class="dt">labels=</span>genes, <span class="dt">pos=</span><span class="dv">3</span>))</code></pre></div>
<p><img src="figure/AA.Rmd/plot_targeted_Tf-1.png" width="700" style="display: block; margin: auto;" /></p>
</div>
<div id="b-top-signals" class="section level2">
<h2>(b) Top Signals</h2>
<div id="i-batf-analysis-with-permutation-test" class="section level3">
<h3>(i) BATF analysis with permutation test</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">batf =<span class="st"> </span><span class="kw">make_target_permtest</span>(<span class="st">&quot;BATF&quot;</span>)
<span class="kw">hist</span>(batf<span class="op">$</span>out, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(batf<span class="op">$</span>d<span class="op">+</span><span class="dv">10</span>, <span class="kw">max</span>(batf<span class="op">$</span>out))),
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;p value = &quot;</span>, batf<span class="op">$</span>p),
     <span class="dt">xlab =</span> <span class="st">&quot;null distribution of d of BATF&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>batf<span class="op">$</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA.Rmd/batf-1.png" width="400" style="display: block; margin: auto;" /></p>
</div>
<div id="ii-znf785-analysis-with-permutation-test" class="section level3">
<h3>(ii) ZNF785 analysis with permutation test</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">znf =<span class="st"> </span><span class="kw">make_target_permtest</span>(<span class="st">&quot;ZNF785&quot;</span>)
<span class="kw">hist</span>(znf<span class="op">$</span>out, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(znf<span class="op">$</span>d<span class="op">+</span><span class="dv">10</span>, <span class="kw">max</span>(znf<span class="op">$</span>out))),
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;p value = &quot;</span>, znf<span class="op">$</span>p),
     <span class="dt">xlab =</span> <span class="st">&quot;null distribution of d of ZNF785&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>znf<span class="op">$</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA.Rmd/znf785-1.png" width="500" style="display: block; margin: auto;" /></p>
</div>
<div id="iii-ncor2-with-targeted-genes-only" class="section level3">
<h3>(iii) NCOR2 with targeted genes only</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ncor =<span class="st"> </span><span class="kw">make_target_permtest</span>(<span class="st">&quot;NCOR2&quot;</span>)
<span class="kw">hist</span>(ncor<span class="op">$</span>out, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(ncor<span class="op">$</span>d<span class="op">+</span><span class="dv">10</span>, <span class="kw">max</span>(ncor<span class="op">$</span>out))),
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;p value = &quot;</span>, ncor<span class="op">$</span>p),
     <span class="dt">xlab =</span> <span class="st">&quot;null distribution of d of NCOR2&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>ncor<span class="op">$</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA.Rmd/ncor_targets-1.png" width="400" style="display: block; margin: auto;" /></p>
We conducted different analyses with the same data sets, and found some overlapping signals that appeared across the different analyses such as NCOR2 and ZEB2. <br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] biomaRt_2.36.1    KEGGREST_1.22.0   ggplot2_3.1.0     nleqslv_3.3.2    
[5] data.table_1.12.0 mvtnorm_1.0-10    MASS_7.3-51.3    

loaded via a namespace (and not attached):
 [1] progress_1.2.0            tidyselect_0.2.5         
 [3] xfun_0.5                  purrr_0.3.2              
 [5] colorspace_1.4-1          htmltools_0.3.6          
 [7] stats4_3.5.3              yaml_2.2.0               
 [9] blob_1.1.1                XML_3.98-1.19            
[11] rlang_0.3.3               pillar_1.3.1             
[13] glue_1.3.1                withr_2.1.2              
[15] DBI_1.0.0                 bit64_0.9-7              
[17] BiocGenerics_0.26.0       plyr_1.8.4               
[19] stringr_1.4.0             zlibbioc_1.26.0          
[21] Biostrings_2.50.2         munsell_0.5.0            
[23] gtable_0.3.0              workflowr_1.2.0          
[25] memoise_1.1.0             evaluate_0.13            
[27] RcppArmadillo_0.8.600.0.0 Biobase_2.40.0           
[29] knitr_1.22                IRanges_2.16.0           
[31] parallel_3.5.3            AnnotationDbi_1.42.1     
[33] Rcpp_1.0.1                scales_1.0.0             
[35] backports_1.1.3           S4Vectors_0.20.1         
[37] XVector_0.20.0            bit_1.1-14               
[39] fs_1.2.7                  hms_0.4.2                
[41] png_0.1-7                 digest_0.6.18            
[43] stringi_1.4.3             dplyr_0.8.0.1            
[45] grid_3.5.3                rprojroot_1.3-2          
[47] tools_3.5.3               bitops_1.0-6             
[49] magrittr_1.5              RSQLite_2.1.1            
[51] lazyeval_0.2.2            RCurl_1.95-4.12          
[53] tibble_2.1.1              crayon_1.3.4             
[55] whisker_0.3-2             pkgconfig_2.0.2          
[57] prettyunits_1.0.2         assertthat_0.2.1         
[59] rmarkdown_1.12            httr_1.4.0               
[61] R6_2.4.0                  git2r_0.25.2             
[63] compiler_3.5.3           </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
