<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Tae Kim" />


<title>AA vs EA</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Differential Coexpression Network</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="AA_EA.html">Discrete Example</a>
</li>
<li>
  <a href="AA.html">Continuous Example</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tk382/networkAnc">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">AA vs EA</h1>
<h4 class="author">Tae Kim</h4>
<h4 class="date">3/23/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-04-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>diffNet/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190331code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190331)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190331code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190331)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-23
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>AA_EA_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtk382networkAnctree606fa41352b1d9963546017abf2429d1985c8575targetblank606fa41a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tk382/networkAnc/tree/606fa41352b1d9963546017abf2429d1985c8575" target="_blank">606fa41</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtk382networkAnctree606fa41352b1d9963546017abf2429d1985c8575targetblank606fa41a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/.DS_Store
    Ignored:    R/.Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store

Untracked files:
    Untracked:  .Rprofile
    Untracked:  .gitignore
    Untracked:  R/not_build/
    Untracked:  analysis/AA_EA_cache/
    Untracked:  analysis/AA_skin_not_sun_exposed.Rmd
    Untracked:  analysis/AA_skin_not_sun_exposed.pdf
    Untracked:  analysis/HLA-DQA1.Rmd
    Untracked:  analysis/HLA-DQA1.pdf
    Untracked:  analysis/HLAgenes.Rmd
    Untracked:  analysis/HLAgenes.pdf
    Untracked:  analysis/HMP.Rmd
    Untracked:  analysis/HMP.pdf
    Untracked:  analysis/KgreaterthanN.Rmd
    Untracked:  analysis/KgreaterthanN.pdf
    Untracked:  analysis/TFs.Rmd
    Untracked:  analysis/TFs.pdf
    Untracked:  analysis/activation_permutation_test_result.Rdata
    Untracked:  analysis/autoimmune_pathway.Rmd
    Untracked:  analysis/binding_permutation_test_result.Rdata
    Untracked:  analysis/catalysis_permutation_test_result.Rdata
    Untracked:  analysis/correction.R
    Untracked:  analysis/degree_simulation.R
    Untracked:  analysis/export_genenames.R
    Untracked:  analysis/expression_permutation_test_result.Rdata
    Untracked:  analysis/inhibition_permutation_test_result.Rdata
    Untracked:  analysis/largeK.Rmd
    Untracked:  analysis/largeK.pdf
    Untracked:  analysis/make_reference_hgnc.R
    Untracked:  analysis/normality_breaks.R
    Untracked:  analysis/permutation_test.Rmd
    Untracked:  analysis/permutation_test.pdf
    Untracked:  analysis/ptmod_permutation_test_result.Rdata
    Untracked:  analysis/reaction_permutation_test_result.Rdata
    Untracked:  analysis/remove_covariates.Rmd
    Untracked:  analysis/score_test_simulation.R
    Untracked:  analysis/simulation1.Rmd
    Untracked:  analysis/simulation2.Rmd
    Untracked:  analysis/skin_AA.Rmd
    Untracked:  analysis/skin_AA.pdf
    Untracked:  analysis/skin_not__sun_exposed.pdf
    Untracked:  analysis/skin_not__sun_exposed_files/
    Untracked:  analysis/skin_not_sun_exposed.Rmd
    Untracked:  analysis/skin_not_sun_exposed.pdf
    Untracked:  analysis/smallk_known_sigma.Rmd
    Untracked:  analysis/smallk_known_sigma.pdf
    Untracked:  analysis/smallk_unknown_sigma.Rmd
    Untracked:  analysis/smallk_unknown_sigma.pdf
    Untracked:  analysis/type1error.Rmd
    Untracked:  code/
    Untracked:  data/
    Untracked:  docs/figure/simulation1.Rmd/
    Untracked:  docs/figure/simulation2.Rmd/
    Untracked:  old_analysis/
    Untracked:  output/
    Untracked:  src/networkAnc.so

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/606fa41352b1d9963546017abf2429d1985c8575/analysis/AA_EA.Rmd" target="_blank">606fa41</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
<td>
wflow_publish(c(“analysis/AA_EA.Rmd”, “analysis/index.Rmd”), republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/AA_EA.html" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/dc21566d1401835d1d813aceac5a26cd6e12d8e8/docs/AA_EA.html" target="_blank">dc21566</a>
</td>
<td>
tk382
</td>
<td>
2019-04-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/368bd3ed13c957720160a6db953a0de64a05ac07/docs/AA_EA.html" target="_blank">368bd3e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/00c00595f30810d0481eb5f9e19924bbdb5248ac/analysis/AA_EA.Rmd" target="_blank">00c0059</a>
</td>
<td>
tk382
</td>
<td>
2019-04-02
</td>
<td>
wflow_publish(c(“analysis/AA.Rmd”, “analysis/AA_EA.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/ccd2ef25139743c5bbcce8ba745b2b8002f22776/docs/AA_EA.html" target="_blank">ccd2ef2</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/0844f8f02f8709617d6cdf3b4d9f007de63bf0b6/analysis/AA_EA.Rmd" target="_blank">0844f8f</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
wflow_publish(“analysis/AA_EA.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/AA_EA.html" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tk382/networkAnc/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/AA_EA.html" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tk382/networkAnc/blob/71c33685bbe4da65b6153a0e986c0320c6f955fc/analysis/AA_EA.Rmd" target="_blank">71c3368</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
<td>
wflow_publish(“analysis/AA_EA.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-pre-processing" class="section level1">
<h1>1. Data pre-processing</h1>
<p>We use the RNA-seq expression level data provided by GTEx, which has been already pre-processed. We choose to study not-sun-exposed suprapubic skin tissue as a pilot study because we expect to see some ancestry effect on the expression level on the skin tissue. Genes were selected based on expression threshold of <span class="math inline">\(&gt;0.1\)</span> RPKM in at least 10 individuals and <span class="math inline">\(\geq\)</span> reads in at least 10 individuals. Then the expression values were quantile normalized to the average empirical distribution observed across samples. That means, each gene has the same distribution across the individuals. Then they were inverse quantile normalized to fit standard normal distribution.</p>
<div id="a-load-and-clean-the-data" class="section level2">
<h2>(a) load and clean the data</h2>
<p>We read the expression level data and match the individuals with the ancestry information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;data/skin_not_sun_exposed_suprapubic.Rdata&quot;</span>)
exp =<span class="st"> </span>skin; <span class="kw">rm</span>(skin)
<span class="kw">colnames</span>(exp) =<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;[.]&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="kw">colnames</span>(exp))
ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colSums</span>(<span class="op">!</span><span class="kw">is.na</span>(exp)) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)
exp =<span class="st"> </span>exp[,<span class="op">-</span>ind]
ind=<span class="st"> </span><span class="kw">which</span>(<span class="kw">rowSums</span>(<span class="op">!</span><span class="kw">is.na</span>(exp)) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)
exp =<span class="st"> </span>exp[<span class="op">-</span>ind, ]
<span class="kw">invisible</span>(<span class="kw">gc</span>())
finalA =<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/finalA.txt&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>)
AAs =<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">colnames</span>(exp), finalA<span class="op">$</span>subject)
EAs =<span class="st"> </span><span class="kw">colnames</span>(exp)[<span class="op">!</span><span class="st"> </span><span class="kw">colnames</span>(exp) <span class="op">%in%</span><span class="st"> </span>finalA<span class="op">$</span>subject]
newA=<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">subject =</span> AAs, <span class="dt">A =</span> finalA[<span class="kw">match</span>(AAs, finalA<span class="op">$</span>subject), <span class="dv">2</span>])
Y =<span class="st"> </span>exp[, <span class="kw">c</span>(AAs, EAs)]
<span class="kw">rm</span>(exp); <span class="kw">invisible</span>(<span class="kw">gc</span>())
EA_A =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">subject =</span> EAs, <span class="dt">A =</span> <span class="dv">0</span>)
X =<span class="st"> </span><span class="kw">rbind</span>(newA, EA_A)
<span class="kw">rm</span>(newA, EA_A, AAs, EAs, ind, finalA)
<span class="kw">rownames</span>(Y) =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(Y), <span class="st">&#39;[.]&#39;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])
Y =<span class="st"> </span><span class="kw">t</span>(Y)
orig =<span class="st"> </span>Y
<span class="kw">dim</span>(Y)</code></pre></div>
<pre><code>[1]   190 25192</code></pre>
</div>
<div id="b-scale-the-data-and-regress-out-the-mean" class="section level2">
<h2>(b) scale the data and regress out the mean</h2>
<p>We code African Americans as 0 and European Americans as 1 to compare group by group coexpression difference. We also consider correlation matrix, so we scale the gene expression level matrix to have variance 1. Then we regress out ancestry from the gene expression level to remove the mean effect.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">orig_A =<span class="st"> </span>X<span class="op">$</span>A
X<span class="op">$</span>A[X<span class="op">$</span>A <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>] =<span class="st"> </span><span class="dv">1</span>
X<span class="op">$</span>A =<span class="st"> </span><span class="kw">scale</span>(X<span class="op">$</span>A) <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)<span class="op">-</span><span class="dv">1</span>)
Y =<span class="st"> </span><span class="kw">resid</span>(<span class="kw">lm</span>(Y<span class="op">~</span>X<span class="op">$</span>A))
Y =<span class="st"> </span><span class="kw">scale</span>(Y) <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">length</span>(X<span class="op">$</span>A)<span class="op">-</span><span class="dv">1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ref_chisq =<span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">B =<span class="st"> </span><span class="dv">5000</span></code></pre></div>
</div>
<div id="c-match-ensembl-id-and-hgnc-symbols" class="section level2">
<h2>(c) match ENSEMBL ID and hgnc symbols</h2>
<p><a href="https://www.genenames.org/download/custom/" class="uri">https://www.genenames.org/download/custom/</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hgnc_names =<span class="st"> </span>hgnc[<span class="kw">match</span>(<span class="kw">colnames</span>(Y), hgnc<span class="op">$</span><span class="st">`</span><span class="dt">Ensembl gene ID</span><span class="st">`</span>), <span class="st">&quot;Approved symbol&quot;</span>]
ref =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">ensembl_gene_id =</span> <span class="kw">colnames</span>(Y), <span class="dt">hgnc_symbol =</span> hgnc_names)
ref<span class="op">$</span>ensembl_gene_id =<span class="st"> </span><span class="kw">as.character</span>(ref<span class="op">$</span>ensembl_gene_id)
ref<span class="op">$</span>hgnc_symbol =<span class="st"> </span><span class="kw">as.character</span>(ref<span class="op">$</span>hgnc_symbol)</code></pre></div>
</div>
</div>
<div id="gene-selection" class="section level1">
<h1>2. Gene selection</h1>
<p>In order to avoid extensive multiple testing, we use prior information in biology.</p>
<div id="a-kegg-pathway" class="section level2">
<h2>(a) KEGG pathway</h2>
<p>We believe the disease pathways may reflect the gene-gene correlation structure or gene modules, so we choose three pathways that are related to either skin tissue or immunity - melonoma, asthma, and autoimmune disease.</p>
<p>For each pathway, we take a group of <span class="math inline">\(K\)</span> genes and for each <span class="math inline">\(k\)</span> in <span class="math inline">\(1, \cdots, K\)</span>, we compute <span class="math inline">\(d_k\)</span> and compare it to the empirical distribution by permuting the ancestry. Then we show qqplot of the p values against the uniform distribution after transformation <span class="math inline">\(-log_{10}(p)\)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get_result_kegg =<span class="st"> </span><span class="cf">function</span>(hsacode, B){
  out =<span class="st"> </span><span class="kw">keggGet</span>(hsacode)
  out =<span class="st"> </span>out[[<span class="dv">1</span>]]
  genes =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(out<span class="op">$</span>GENE, <span class="st">&quot;;&quot;</span>), <span class="cf">function</span>(x) x[<span class="dv">1</span>])
  genes =<span class="st"> </span>genes[<span class="kw">seq</span>(<span class="dv">2</span>,<span class="dv">144</span>,<span class="dt">by=</span><span class="dv">2</span>)]
  results =<span class="st"> </span>ref[<span class="kw">match</span>(genes, ref<span class="op">$</span>hgnc_symbol), ]
  genes =<span class="st"> </span><span class="kw">colnames</span>(Y)[<span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id)]
  results =<span class="st"> </span>results[<span class="kw">match</span>(genes, results<span class="op">$</span>ensembl_gene_id), ]
  R =<span class="st"> </span>Y[,genes]
  d =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">ncol</span>(R))
  p =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">ncol</span>(R))
  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(R)){
    d[i] =<span class="st"> </span><span class="kw">get_degree_c</span>(X<span class="op">$</span>A, R[,i], R[,<span class="op">-</span>i])
    storeW =<span class="st"> </span><span class="kw">store_W_c</span>(R[,i], R[,<span class="op">-</span>i])
    out =<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">bootstrap_c</span>(X<span class="op">$</span>A, B, storeW))
    p[i] =<span class="st"> </span><span class="kw">sum</span>(out <span class="op">&gt;</span><span class="st"> </span>d[i]) <span class="op">/</span><span class="st"> </span>B
  }
  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">p =</span> p, <span class="dt">d =</span> d, <span class="dt">genes =</span> results))
}

out =<span class="st"> </span><span class="kw">get_result_kegg</span>(<span class="st">&quot;hsa05218&quot;</span>, B) <span class="co">#melanoma disease</span>
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">qqplot</span>(<span class="op">-</span><span class="kw">log</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="kw">length</span>(out<span class="op">$</span>p))), <span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p),
     <span class="dt">xlab =</span> <span class="st">&#39;-log uniform&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;-log p&#39;</span>,
     <span class="dt">main =</span> <span class="st">&quot;melanoma&quot;</span>)
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)

df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">p =</span> <span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p), <span class="dt">genes =</span> out<span class="op">$</span>genes<span class="op">$</span>hgnc_symbol, <span class="dt">ind =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(out<span class="op">$</span>p))
<span class="kw">plot</span>(<span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p), 
     <span class="dt">ylab =</span> <span class="st">&#39;-log p&#39;</span>, 
     <span class="dt">main =</span> <span class="st">&#39;-log(p) values&#39;</span>,
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">length</span>(out<span class="op">$</span>p)<span class="op">+</span><span class="dv">10</span>))
ind =<span class="st"> </span><span class="kw">which</span>(df<span class="op">$</span>p <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>)
df =<span class="st"> </span>df[ind, ]
<span class="kw">with</span>(df, <span class="kw">text</span>(p <span class="op">~</span><span class="st"> </span>ind, <span class="dt">labels =</span> genes, <span class="dt">pos =</span> <span class="dv">4</span>))</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-4-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">out =<span class="st"> </span><span class="kw">get_result_kegg</span>(<span class="st">&quot;hsa05310&quot;</span>, B) <span class="co">#asthma</span>
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">qqplot</span>(<span class="op">-</span><span class="kw">log</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="kw">length</span>(out<span class="op">$</span>p))), <span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p),
     <span class="dt">xlab =</span> <span class="st">&#39;-log uniform&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;-log p&#39;</span>,
     <span class="dt">main =</span> <span class="st">&quot;asthma&quot;</span>)
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)

df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">p =</span> <span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p), <span class="dt">genes =</span> out<span class="op">$</span>genes<span class="op">$</span>hgnc_symbol, <span class="dt">ind =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(out<span class="op">$</span>p))
<span class="kw">plot</span>(<span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p), 
     <span class="dt">ylab =</span> <span class="st">&#39;-log p&#39;</span>, 
     <span class="dt">main =</span> <span class="st">&#39;-log(p) values&#39;</span>,
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">length</span>(out<span class="op">$</span>p)<span class="op">+</span><span class="dv">10</span>))
ind =<span class="st"> </span><span class="kw">which</span>(df<span class="op">$</span>p <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>)
df =<span class="st"> </span>df[ind, ]
<span class="kw">with</span>(df, <span class="kw">text</span>(p <span class="op">~</span><span class="st"> </span>ind, <span class="dt">labels =</span> genes, <span class="dt">pos =</span> <span class="dv">4</span>))</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-4-2.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-4-2.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">out =<span class="st"> </span><span class="kw">get_result_kegg</span>(<span class="st">&quot;hsa05320&quot;</span>, B) <span class="co">#autoimmune</span>
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">qqplot</span>(<span class="op">-</span><span class="kw">log</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="kw">length</span>(out<span class="op">$</span>p))), <span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p),
     <span class="dt">xlab =</span> <span class="st">&#39;-log uniform&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;-log p&#39;</span>,
     <span class="dt">main =</span> <span class="st">&quot;auto immune&quot;</span>)
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)

df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">p =</span> <span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p), <span class="dt">genes =</span> out<span class="op">$</span>genes<span class="op">$</span>hgnc_symbol, <span class="dt">ind =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(out<span class="op">$</span>p))
<span class="kw">plot</span>(<span class="op">-</span><span class="kw">log</span>(out<span class="op">$</span>p), 
     <span class="dt">ylab =</span> <span class="st">&#39;-log p&#39;</span>, 
     <span class="dt">main =</span> <span class="st">&#39;-log(p) values&#39;</span>,
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">length</span>(out<span class="op">$</span>p)<span class="op">+</span><span class="dv">10</span>))
ind =<span class="st"> </span><span class="kw">which</span>(df<span class="op">$</span>p <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>)
df =<span class="st"> </span>df[ind, ]
<span class="kw">with</span>(df, <span class="kw">text</span>(p <span class="op">~</span><span class="st"> </span>ind, <span class="dt">labels =</span> genes, <span class="dt">pos =</span> <span class="dv">4</span>))</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-4-3.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-3">
Past versions of unnamed-chunk-4-3.png
</button>
</p>
<div id="fig-unnamed-chunk-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-4-3.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>

</div>
<div id="b-hla-genes" class="section level2">
<h2>(b) HLA genes</h2>
<p>We know that HLA genes are connected to many other genes and play critical role in regulating the gene transcription. We obtain the list of HLA genes and compute <span class="math inline">\(d_1\)</span> by adding up pair-wise test statistics across the entire genome.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hla_genes =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;HLA-A&quot;</span>, <span class="st">&quot;HLA-B&quot;</span>, <span class="st">&quot;HLA-C&quot;</span>, <span class="st">&quot;HLA-E&quot;</span>, 
              <span class="st">&quot;HLA-F&quot;</span>, <span class="st">&quot;HLA-G&quot;</span>, <span class="st">&quot;HLA-H&quot;</span>, <span class="st">&quot;HLA-J&quot;</span>, 
              <span class="st">&quot;HLA-K&quot;</span>, <span class="st">&quot;HLA-L&quot;</span>, <span class="st">&quot;HLA-N&quot;</span>, <span class="st">&quot;HLA-P&quot;</span>, 
              <span class="st">&quot;HLA-S&quot;</span>, <span class="st">&quot;HLA-T&quot;</span>, <span class="st">&quot;HLA-U&quot;</span>, <span class="st">&quot;HLA-V&quot;</span>, 
              <span class="st">&quot;HLA-W&quot;</span>, <span class="st">&quot;HLA-X&quot;</span>, <span class="st">&quot;HLA-Y&quot;</span>, <span class="st">&quot;HLA-Z&quot;</span>,
              <span class="st">&quot;HLA-DRA&quot;</span>, <span class="st">&quot;HLA-DRB2&quot;</span>, <span class="st">&quot;HLA-DRB3&quot;</span>,
              <span class="st">&quot;HLA-DRB4&quot;</span>, <span class="st">&quot;HLA-DRB5&quot;</span>, <span class="st">&quot;HLA-DRB6&quot;</span>,
              <span class="st">&quot;HLA-DRB7&quot;</span>, <span class="st">&quot;HLA-DRB8&quot;</span>, <span class="st">&quot;HLA-DRB9&quot;</span>,
              <span class="st">&quot;HLA-DQA1&quot;</span>, <span class="st">&quot;HLA-DQB1&quot;</span>, <span class="st">&quot;HLA-DQA2&quot;</span>,
              <span class="st">&quot;HLA-DQB2&quot;</span>, <span class="st">&quot;HLA-DQB3&quot;</span>, <span class="st">&quot;HLA-DOA&quot;</span>,
              <span class="st">&quot;HLA-DMA&quot;</span>, <span class="st">&quot;HLA-DMB&quot;</span>, <span class="st">&quot;HLA-DPA1&quot;</span>, <span class="st">&quot;HLA-DPB1&quot;</span>,
              <span class="st">&quot;HLA-DPA2&quot;</span>, <span class="st">&quot;HLA-DPB2&quot;</span>, <span class="st">&quot;HLA-DPA3&quot;</span>,
              <span class="st">&quot;HFE&quot;</span>, <span class="st">&quot;TAP1&quot;</span>, <span class="st">&quot;TAP2&quot;</span>, <span class="st">&quot;PSMB9&quot;</span>, <span class="st">&quot;PSMB8&quot;</span>,
              <span class="st">&quot;MICA&quot;</span>, <span class="st">&quot;MICB&quot;</span>, <span class="st">&quot;MICC&quot;</span>, <span class="st">&quot;MICD&quot;</span>, <span class="st">&quot;MICE&quot;</span>)

results =<span class="st"> </span>ref[<span class="kw">match</span>(hla_genes, ref<span class="op">$</span>hgnc_symbol), ]

genes =<span class="st"> </span><span class="kw">colnames</span>(Y)[<span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id)]
results =<span class="st"> </span>results[<span class="kw">match</span>(genes, results<span class="op">$</span>ensembl_gene_id), ]

scores =<span class="st"> </span>newscores =<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="kw">length</span>(genes), <span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>)
coef =<span class="st"> </span><span class="kw">cubic_coeff_c</span>(X<span class="op">$</span>A, <span class="kw">qchisq</span>(<span class="dv">1</span><span class="op">-</span><span class="fl">0.05</span><span class="op">/</span><span class="kw">ncol</span>(Y), <span class="dv">1</span>), <span class="dv">2</span>, <span class="dv">2</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genes)){
  ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">==</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id[i])
  W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,ind], Y[, <span class="op">-</span>ind])
  scores[i,] =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W)
  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(scores)){
    roots =<span class="st"> </span><span class="kw">polyroot</span>(<span class="kw">c</span>(<span class="op">-</span>scores[i,j], coef))
    newscores[i,j] =<span class="st"> </span><span class="kw">Re</span>(roots)[<span class="kw">abs</span>(<span class="kw">Im</span>(roots)) <span class="op">&lt;</span><span class="st"> </span><span class="fl">1e-6</span>]
  }
}</code></pre></div>
<p>And the result is plotted below, where the red line is the expected mean. HLA-DQA1 stands out in its test statistic.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
d =<span class="st"> </span><span class="kw">rowSums</span>(newscores)
df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">d =</span> d, <span class="dt">genes =</span> results<span class="op">$</span>hgnc_symbol, <span class="dt">xaxis =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(d))
<span class="kw">plot</span>(d, 
     <span class="dt">ylab =</span> <span class="st">&#39;d&#39;</span>, 
     <span class="dt">main =</span> <span class="st">&#39;HLA-genes&#39;</span>,
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">45</span>),
     <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(d) <span class="op">+</span><span class="st"> </span><span class="dv">100</span>))
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">25190</span>, <span class="dt">col=</span><span class="st">&#39;red&#39;</span>)
ind =<span class="st"> </span><span class="kw">which</span>(d <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span><span class="op">*</span><span class="kw">ncol</span>(Y))
df =<span class="st"> </span>df[ind, ]
<span class="kw">with</span>(df, <span class="kw">text</span>(d <span class="op">~</span><span class="st"> </span>xaxis, <span class="dt">labels =</span> genes, <span class="dt">pos =</span> <span class="dv">4</span>))

ref_chisq =<span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length=</span><span class="kw">length</span>(newscores[<span class="dv">1</span>,])), <span class="dv">1</span>)
ind =<span class="st"> </span><span class="kw">which</span>(results<span class="op">$</span>hgnc_symbol <span class="op">==</span><span class="st"> &quot;HLA-DQA1&quot;</span>)
<span class="kw">qqplot</span>(ref_chisq, newscores[ind,],  <span class="dt">main =</span> <span class="st">&#39;HLA-DQA1 scores&#39;</span>, <span class="dt">ylab=</span><span class="st">&#39;scores&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-6-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-6-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="c-transcription-factors" class="section level2">
<h2>(c) Transcription Factors</h2>
<div id="i-read-tf-data" class="section level3">
<h3>(i) read TF data</h3>
<p>Since transcription factors usually regulate the transcription of multiple genes, we thought they were appropriate targets to study their variance in activity across different ancestry.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">TF =<span class="st"> </span><span class="kw">fread</span>(<span class="st">&quot;data/not_build/TFcheckpoint.txt&quot;</span>, <span class="dt">header=</span><span class="ot">TRUE</span>)
TF =<span class="st"> </span><span class="kw">setDF</span>(TF)
TF =<span class="st"> </span>TF[TF<span class="op">$</span>TFClass_human <span class="op">==</span><span class="st"> &quot;TFclass&quot;</span>, ] <span class="co">#newly added!</span>
TF =<span class="st"> </span>TF[,<span class="dv">1</span>]
results =<span class="st"> </span>ref[<span class="kw">match</span>(TF, ref<span class="op">$</span>hgnc_symbol), ]

genes =<span class="st"> </span><span class="kw">colnames</span>(Y)[<span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id)]
results =<span class="st"> </span>results[<span class="kw">match</span>(genes, results<span class="op">$</span>ensembl_gene_id), ]</code></pre></div>
</div>
<div id="ii-test-all-the-tfs" class="section level3">
<h3>(ii) test all the TFs</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
scores =<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="kw">length</span>(genes), <span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genes)){
  ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">==</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id[i])
  W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,ind], Y[, <span class="op">-</span>ind])
  scores[i,] =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W)
}

d =<span class="st"> </span><span class="kw">rowSums</span>(scores)
df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">d =</span> d, <span class="dt">genes =</span> results<span class="op">$</span>hgnc_symbol, <span class="dt">xaxis =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(d))
<span class="kw">plot</span>(d, 
     <span class="dt">ylab =</span> <span class="st">&#39;d&#39;</span>, 
     <span class="dt">main =</span> <span class="st">&#39;Transcription Factors&#39;</span>)
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">25190</span>, <span class="dt">col=</span><span class="st">&#39;red&#39;</span>)
ind =<span class="st"> </span><span class="kw">which</span>(d <span class="op">&gt;</span><span class="st"> </span><span class="kw">ncol</span>(Y) <span class="op">+</span><span class="st"> </span><span class="dv">200</span> <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="dv">2</span><span class="op">*</span><span class="kw">ncol</span>(Y)))
df =<span class="st"> </span>df[ind, ]
<span class="kw">with</span>(df, <span class="kw">text</span>(d <span class="op">~</span><span class="st"> </span>xaxis, <span class="dt">labels =</span> genes, <span class="dt">pos =</span> <span class="dv">4</span>))

ref_chisq =<span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>, <span class="dt">length=</span><span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(genes)){
    <span class="cf">if</span>(results<span class="op">$</span>hgnc_symbol[i] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;FOXM1&quot;</span>)){
      <span class="kw">qqplot</span>(ref_chisq, scores[i, ], 
            <span class="dt">main =</span> <span class="kw">paste</span>(results<span class="op">$</span>hgnc_symbol[i], <span class="st">&#39;unadjusted&#39;</span>),
            <span class="dt">cex =</span> <span class="fl">0.5</span>)
      <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
  }
}</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-8-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-8-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-8-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>

</div>
</div>
</div>
<div id="investigating-the-top-signals" class="section level1">
<h1>3. Investigating the top signals</h1>
<div id="a-hla-dqa1" class="section level2">
<h2>(a) HLA-DQA1</h2>
<div id="i-score-test" class="section level3">
<h3>(i) Score test</h3>
<p>We re-cap what we have observed in HLA-DQA1 for the score tests.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hla_genes =<span class="st"> &quot;HLA-DQA1&quot;</span>
results =<span class="st"> </span>ref[<span class="kw">match</span>(hla_genes, ref<span class="op">$</span>hgnc_symbol), ]

dqa_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(results<span class="op">$</span>ensembl_gene_id))
smallY =<span class="st"> </span>Y[,<span class="op">-</span>dqa_ind]
s =<span class="st"> </span>news =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>)){
  s[i] =<span class="st"> </span><span class="kw">get_score_c</span>(X<span class="op">$</span>A, Y[,dqa_ind], smallY[,i])
}
<span class="kw">plot</span>(<span class="kw">sort</span>(s) <span class="op">~</span><span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length=</span><span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span>),
     <span class="dt">type =</span> <span class="st">&#39;l&#39;</span>,
     <span class="dt">main =</span> <span class="st">&#39;pairwise scores&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;reference chi1&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;sorted scores&#39;</span>); 
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-9-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-9-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/figure/AA_EA.Rmd/unnamed-chunk-9-1.png" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-9-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ii-permutation-test" class="section level3">
<h3>(ii) Permutation test</h3>
<p>In order to verify if HLA-DQA truly has a signal, we perform permutation test. The p-value is indeed very small.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,dqa_ind], Y[, <span class="op">-</span>dqa_ind])
dqa_scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W)
dqa_d =<span class="st"> </span><span class="kw">get_degree_c</span>(X<span class="op">$</span>A, Y[,dqa_ind], Y[,<span class="op">-</span>dqa_ind])
<span class="co"># shuffled_A = shuffle_x_c(X$A, B)</span>
storeW =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,dqa_ind], Y[,<span class="op">-</span>dqa_ind])
out =<span class="st"> </span><span class="kw">bootstrap_c</span>(X<span class="op">$</span>A, B, storeW)
bootstrapped_d =<span class="st"> </span><span class="kw">rowSums</span>(out)
dqa_p =<span class="st"> </span><span class="kw">sum</span>(bootstrapped_d <span class="op">&gt;</span><span class="st"> </span>dqa_d) <span class="op">/</span><span class="st"> </span>B
<span class="kw">hist</span>(bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">10000</span>, <span class="kw">max</span>(<span class="kw">max</span>(bootstrapped_d), dqa_d))<span class="op">+</span><span class="dv">20</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>dqa_d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-10-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-10-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(dqa_p)</code></pre></div>
<pre><code>[1] 0.007</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># par(mfrow = c(2,3))</span>
<span class="co"># for (i in 1:12){</span>
<span class="co">#   qqplot(ref_chisq, out[i, ], cex = 0.5, type = &#39;l&#39;, ylab = &#39;permuted score&#39;)</span>
<span class="co">#   abline(0,1,col = &#39;red&#39;)</span>
<span class="co"># }</span></code></pre></div>
</div>
<div id="iii-genes-with-the-highest-scores" class="section level3">
<h3>(iii) Genes with the highest scores</h3>
<p>We make some diagnosis plot to observe what is driving the signal.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">max_scores =<span class="st"> </span><span class="kw">sort</span>(s, <span class="dt">decreasing=</span><span class="ot">TRUE</span>, <span class="dt">index.return=</span><span class="ot">TRUE</span>)
results =<span class="st"> </span>ref[<span class="kw">match</span>(<span class="kw">colnames</span>(Y[,<span class="op">-</span>dqa_ind])[max_scores<span class="op">$</span>ix[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]], ref<span class="op">$</span>ensembl_gene_id), ]
results =<span class="st"> </span>results[<span class="kw">match</span>(results<span class="op">$</span>ensembl_gene_id, <span class="kw">colnames</span>(Y[,<span class="op">-</span>dqa_ind])[max_scores<span class="op">$</span>ix[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]]), ]
top_genes =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">gene_name =</span> results<span class="op">$</span>hgnc_symbol, <span class="dt">score =</span> max_scores<span class="op">$</span>x[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>],
                <span class="dt">ensembl =</span> results<span class="op">$</span>ensembl_gene_id)
<span class="co"># print(top_genes)</span>
AA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)
EA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)
diff_cor =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> top_genes<span class="op">$</span>gene_name,
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,dqa_ind], 
                                          (orig[AA_ind,<span class="op">-</span>dqa_ind])[, max_scores<span class="op">$</span>ix[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]])),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[EA_ind,dqa_ind], 
                                          (orig[EA_ind,<span class="op">-</span>dqa_ind])[, max_scores<span class="op">$</span>ix[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]])))
<span class="kw">print</span>(<span class="kw">cbind</span>(top_genes, diff_cor[,<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>]))</code></pre></div>
<pre><code>    gene_name    score         ensembl        AA          EA
1        PKP4 28.88540 ENSG00000144283 0.3478526 -0.14944539
2       UPK1B 26.21919 ENSG00000114638 0.4189482 -0.10035988
3      OSBPL6 24.57978 ENSG00000079156 0.4538671 -0.12932814
4        &lt;NA&gt; 23.70785 ENSG00000272008 0.4588957 -0.15284831
5  SUCLA2-AS1 23.53029 ENSG00000227848 0.3425440 -0.08661835
6       SUDS3 23.45660 ENSG00000111707 0.4111445 -0.09436705
7       EIF4B 23.05234 ENSG00000063046 0.2536646 -0.01610219
8       PRSS3 23.04438 ENSG00000010438 0.2267855 -0.14215591
9        LRBA 22.55396 ENSG00000198589 0.4782435  0.04157245
10    PLA2G4E 21.76423 ENSG00000188089 0.1836475 -0.22015849</code></pre>
</div>
<div id="iv-the-expression-level-of-hla-dqa1s-top-signal-lrba" class="section level3">
<h3>(iv) The expression level of HLA-DQA1’s top signal LRBA</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">AA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)
EA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)
diff_cor =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> top_genes<span class="op">$</span>gene_name,
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,dqa_ind], 
                                          (orig[AA_ind,<span class="op">-</span>dqa_ind])[, max_scores<span class="op">$</span>ix[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]])),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[EA_ind,dqa_ind], 
                                          (orig[EA_ind,<span class="op">-</span>dqa_ind])[, max_scores<span class="op">$</span>ix[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]])))
<span class="kw">print</span>(diff_cor)</code></pre></div>
<pre><code>        genes        AA          EA
1        PKP4 0.3478526 -0.14944539
2       UPK1B 0.4189482 -0.10035988
3      OSBPL6 0.4538671 -0.12932814
4        &lt;NA&gt; 0.4588957 -0.15284831
5  SUCLA2-AS1 0.3425440 -0.08661835
6       SUDS3 0.4111445 -0.09436705
7       EIF4B 0.2536646 -0.01610219
8       PRSS3 0.2267855 -0.14215591
9        LRBA 0.4782435  0.04157245
10    PLA2G4E 0.1836475 -0.22015849</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(orig[AA_ind, dqa_ind] <span class="op">~</span><span class="st"> </span>orig[AA_ind,<span class="op">-</span>dqa_ind][, max_scores<span class="op">$</span>ix[<span class="dv">1</span>]],
     <span class="dt">main =</span> <span class="st">&#39;African Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;LRBA&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;HLA-DQA1&#39;</span>)
<span class="kw">plot</span>(orig[EA_ind, dqa_ind] <span class="op">~</span><span class="st"> </span>orig[EA_ind,<span class="op">-</span>dqa_ind][, max_scores<span class="op">$</span>ix[<span class="dv">1</span>]],
     <span class="dt">main =</span> <span class="st">&#39;European Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;LRBA&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;HLA-DQA1&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-12-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-12-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/figure/AA_EA.Rmd/unnamed-chunk-12-1.png" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-12-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="v-box-plot" class="section level3">
<h3>(v) Box Plot</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">AA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)
EA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)
dqa_exp =<span class="st"> </span><span class="kw">as.matrix</span>(orig[,dqa_ind])
dqa_exp_aa =<span class="st"> </span>dqa_exp[AA_ind]
dqa_exp_ea =<span class="st"> </span>dqa_exp[EA_ind]

df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">exp =</span> dqa_exp, <span class="dt">label =</span> <span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;African_Amsericans&quot;</span>, <span class="kw">length</span>(AA_ind)),<span class="kw">rep</span>(<span class="st">&quot;European_Americans&quot;</span>, <span class="kw">length</span>(EA_ind))))

<span class="kw">boxplot</span>(exp <span class="op">~</span><span class="st"> </span>label, <span class="dt">data =</span> df, <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">main =</span> <span class="st">&quot;distribution of expression of HLA-DQA1&quot;</span>)
<span class="kw">stripchart</span>(exp <span class="op">~</span><span class="st"> </span>label, <span class="dt">vertical =</span> <span class="ot">TRUE</span>, <span class="dt">data =</span> df, 
    <span class="dt">method =</span> <span class="st">&quot;jitter&quot;</span>, <span class="dt">add =</span> <span class="ot">TRUE</span>, <span class="dt">pch =</span> <span class="dv">4</span>, <span class="dt">col =</span> <span class="st">&#39;blue&#39;</span>, <span class="dt">cex=</span><span class="fl">0.5</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-13-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-13-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-13-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>

</div>
</div>
<div id="b-foxm1" class="section level2">
<h2>(b) FOXM1</h2>
<div id="i-score-test-1" class="section level3">
<h3>(i) Score test</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gene =<span class="st"> &quot;FOXM1&quot;</span>
results =<span class="st"> </span>ref[<span class="kw">match</span>(gene, ref<span class="op">$</span>hgnc_symbol), ]

fox_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(results<span class="op">$</span>ensembl_gene_id))
smallY =<span class="st"> </span>Y[,<span class="op">-</span>fox_ind]
s =<span class="st"> </span>news =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>)){
  s[i] =<span class="st"> </span><span class="kw">get_score_c</span>(X<span class="op">$</span>A, Y[,fox_ind], smallY[,i])
}
<span class="kw">plot</span>(<span class="kw">sort</span>(s) <span class="op">~</span><span class="st"> </span><span class="kw">qchisq</span>(<span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length=</span><span class="kw">ncol</span>(Y)<span class="op">-</span><span class="dv">1</span>), <span class="dv">1</span>), <span class="dt">type =</span> <span class="st">&#39;l&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>, <span class="dt">main =</span> <span class="st">&#39;adjusted&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-14-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-14-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/figure/AA_EA.Rmd/unnamed-chunk-14-1.png" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-14-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ii-permutation-test-1" class="section level3">
<h3>(ii) Permutation test</h3>
<p>We conduct the permutation test to see if the signal is real.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fox_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span>results[results<span class="op">$</span>hgnc_symbol<span class="op">==</span><span class="st">&quot;FOXM1&quot;</span>, <span class="dv">1</span>])
W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,fox_ind], Y[, <span class="op">-</span>fox_ind])
fox_scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W)
fox_d =<span class="st"> </span><span class="kw">get_degree_c</span>(X<span class="op">$</span>A, Y[,fox_ind], Y[,<span class="op">-</span>fox_ind])
shuffled_A =<span class="st"> </span><span class="kw">shuffle_x_c</span>(X<span class="op">$</span>A, B)
storeW =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,fox_ind], Y[,<span class="op">-</span>fox_ind])
out =<span class="st"> </span><span class="kw">bootstrap_c</span>(X<span class="op">$</span>A, B, storeW)
bootstrapped_d =<span class="st"> </span><span class="kw">rowSums</span>(out)
fox_p =<span class="st"> </span><span class="kw">sum</span>(bootstrapped_d <span class="op">&gt;</span><span class="st"> </span>fox_d) <span class="op">/</span><span class="st"> </span>B
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))
<span class="kw">hist</span>(bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">10000</span>, <span class="kw">max</span>(<span class="kw">max</span>(bootstrapped_d), fox_d))<span class="op">+</span><span class="dv">100</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>fox_d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-15-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-15-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/figure/AA_EA.Rmd/unnamed-chunk-15-1.png" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-15-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(fox_p)</code></pre></div>
<pre><code>[1] 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># par(mfrow = c(2,3))</span>
<span class="co"># for (i in 1:12){</span>
<span class="co">#   qqplot(ref_chisq, out[i, ], cex = 0.5, type = &#39;l&#39;, ylab = &#39;permuted score&#39;)</span>
<span class="co">#   abline(0,1,col = &#39;red&#39;)</span>
<span class="co"># }</span></code></pre></div>
</div>
<div id="iii-box-plot" class="section level3">
<h3>(iii) Box plot</h3>
<p>We inspect the raw data to see any patterns between Afircan Americans and European Americans.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">AA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)
EA_ind =<span class="st"> </span><span class="kw">which</span>(orig_A <span class="op">&lt;=</span><span class="st"> </span><span class="dv">0</span>)
fox_exp =<span class="st"> </span><span class="kw">as.matrix</span>(orig[,fox_ind])
fox_exp_aa =<span class="st"> </span>fox_exp[AA_ind]
fox_exp_ea =<span class="st"> </span>fox_exp[EA_ind]

df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">exp =</span> fox_exp, <span class="dt">label =</span> <span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;African Americans&quot;</span>, <span class="kw">length</span>(AA_ind)),<span class="kw">rep</span>(<span class="st">&quot;European Americans&quot;</span>, <span class="kw">length</span>(EA_ind))))

<span class="kw">boxplot</span>(exp <span class="op">~</span><span class="st"> </span>label, <span class="dt">data =</span> df, <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">main =</span> <span class="st">&quot;distribution of expression of FOXM1&quot;</span>)
<span class="kw">stripchart</span>(exp <span class="op">~</span><span class="st"> </span>label, <span class="dt">vertical =</span> <span class="ot">TRUE</span>, <span class="dt">data =</span> df, 
    <span class="dt">method =</span> <span class="st">&quot;jitter&quot;</span>, <span class="dt">add =</span> <span class="ot">TRUE</span>, <span class="dt">pch =</span> <span class="dv">4</span>, <span class="dt">col =</span> <span class="st">&#39;blue&#39;</span>, <span class="dt">cex=</span><span class="fl">0.5</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-16-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-16-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-16-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="c-correlation-of-hla-dqa1-and-foxm1-against-all-other-genes" class="section level2">
<h2>(c) Correlation of HLA-DQA1 and FOXM1 against all other genes</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">diff_cor_fox =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> <span class="kw">colnames</span>(orig)[<span class="op">-</span>fox_ind],
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,fox_ind], 
                                          (orig[AA_ind,<span class="op">-</span>fox_ind]))),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[EA_ind,fox_ind], 
                                          (orig[EA_ind,<span class="op">-</span>fox_ind]))))
diff_cor_dqa =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> <span class="kw">colnames</span>(orig)[<span class="op">-</span>dqa_ind],
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,dqa_ind], 
                                          (orig[AA_ind,<span class="op">-</span>dqa_ind]))),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[EA_ind,dqa_ind], 
                                          (orig[EA_ind,<span class="op">-</span>dqa_ind]))))
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(diff_cor_dqa<span class="op">$</span>AA <span class="op">~</span><span class="st"> </span>diff_cor_dqa<span class="op">$</span>EA, <span class="dt">cex=</span><span class="fl">0.3</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">main=</span><span class="st">&#39;HLA-DQA1&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;African Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;European Americans&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)

<span class="kw">plot</span>(diff_cor_fox<span class="op">$</span>AA <span class="op">~</span><span class="st"> </span>diff_cor_fox<span class="op">$</span>EA, <span class="dt">cex=</span><span class="fl">0.3</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">main=</span><span class="st">&#39;FOXM1&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;African Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;European Americans&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-17-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-17-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/figure/AA_EA.Rmd/unnamed-chunk-17-1.png" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-17-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="d-what-happens-when-i-reduce-the-sample-size-of-european-americans" class="section level2">
<h2>(d) What happens when I reduce the sample size of European Americans?</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">EA_subset =<span class="st"> </span><span class="kw">sample</span>(EA_ind, <span class="dt">size=</span><span class="dv">31</span>)

diff_cor_fox =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> <span class="kw">colnames</span>(orig)[<span class="op">-</span>fox_ind],
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,fox_ind], 
                                          (orig[AA_ind,<span class="op">-</span>fox_ind]))),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[EA_subset,fox_ind], 
                                          (orig[EA_subset,<span class="op">-</span>fox_ind]))))
diff_cor_dqa =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> <span class="kw">colnames</span>(orig)[<span class="op">-</span>dqa_ind],
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,dqa_ind], 
                                          (orig[AA_ind,<span class="op">-</span>dqa_ind]))),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[EA_subset,dqa_ind], 
                                          (orig[EA_subset,<span class="op">-</span>dqa_ind]))))
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(diff_cor_dqa<span class="op">$</span>AA <span class="op">~</span><span class="st"> </span>diff_cor_dqa<span class="op">$</span>EA, <span class="dt">cex=</span><span class="fl">0.3</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">main=</span><span class="st">&#39;HLA-DQA1&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;African Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;European Americans&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)

<span class="kw">plot</span>(diff_cor_fox<span class="op">$</span>AA <span class="op">~</span><span class="st"> </span>diff_cor_fox<span class="op">$</span>EA, <span class="dt">cex=</span><span class="fl">0.3</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">main=</span><span class="st">&#39;FOXM1&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;African Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;European Americans&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-18-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-18-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/figure/AA_EA.Rmd/unnamed-chunk-18-1.png" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-18-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="e-repeat-hla-dqa1-with-sampled-euopreans" class="section level2">
<h2>(e) Repeat HLA-DQA1 with sampled Euopreans</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[<span class="kw">c</span>(AA_ind, EA_subset),dqa_ind], Y[<span class="kw">c</span>(AA_ind, EA_subset), <span class="op">-</span>dqa_ind])
dqa_scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A[<span class="kw">c</span>(AA_ind, EA_subset)], W)
dqa_d =<span class="st"> </span><span class="kw">get_degree_c</span>(X<span class="op">$</span>A[<span class="kw">c</span>(AA_ind, EA_subset)], Y[<span class="kw">c</span>(AA_ind, EA_subset),dqa_ind], Y[<span class="kw">c</span>(AA_ind, EA_subset),<span class="op">-</span>dqa_ind])
storeW =<span class="st"> </span><span class="kw">store_W_c</span>(Y[<span class="kw">c</span>(AA_ind, EA_subset),dqa_ind], Y[<span class="kw">c</span>(AA_ind, EA_subset),<span class="op">-</span>dqa_ind])
out =<span class="st"> </span><span class="kw">bootstrap_c</span>(X<span class="op">$</span>A[<span class="kw">c</span>(AA_ind, EA_subset)], B, storeW)
bootstrapped_d =<span class="st"> </span><span class="kw">rowSums</span>(out)
dqa_p =<span class="st"> </span><span class="kw">sum</span>(bootstrapped_d <span class="op">&gt;</span><span class="st"> </span>dqa_d) <span class="op">/</span><span class="st"> </span>B
<span class="kw">hist</span>(bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">10000</span>, <span class="kw">max</span>(<span class="kw">max</span>(bootstrapped_d), dqa_d))<span class="op">+</span><span class="dv">20</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>dqa_d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-19-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-19-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/b8ce5bb09c800b6f1973eda359fbae59b7fc4e52/docs/figure/AA_EA.Rmd/unnamed-chunk-19-1.png" target="_blank">b8ce5bb</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/1f3a02e93dbafa6e6ca9ee5ee283ed72ad4965f8/docs/figure/AA_EA.Rmd/unnamed-chunk-19-1.png" target="_blank">1f3a02e</a>
</td>
<td>
tk382
</td>
<td>
2019-04-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(dqa_p)</code></pre></div>
<pre><code>[1] 0.1684</code></pre>
</div>
<div id="f-repeat-fox-m1-with-sampled-euopreans" class="section level2">
<h2>(f) Repeat FOX-M1 with sampled Euopreans</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[<span class="kw">c</span>(AA_ind, EA_subset),fox_ind], Y[<span class="kw">c</span>(AA_ind, EA_subset), <span class="op">-</span>fox_ind])
fox_scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A[<span class="kw">c</span>(AA_ind, EA_subset)], W)
fox_d =<span class="st"> </span><span class="kw">get_degree_c</span>(X<span class="op">$</span>A[<span class="kw">c</span>(AA_ind, EA_subset)], Y[<span class="kw">c</span>(AA_ind, EA_subset),fox_ind], Y[<span class="kw">c</span>(AA_ind, EA_subset),<span class="op">-</span>fox_ind])
storeW =<span class="st"> </span><span class="kw">store_W_c</span>(Y[<span class="kw">c</span>(AA_ind, EA_subset),fox_ind], Y[<span class="kw">c</span>(AA_ind, EA_subset),<span class="op">-</span>fox_ind])
out =<span class="st"> </span><span class="kw">bootstrap_c</span>(X<span class="op">$</span>A[<span class="kw">c</span>(AA_ind, EA_subset)], B, storeW)
bootstrapped_d =<span class="st"> </span><span class="kw">rowSums</span>(out)
fox_p =<span class="st"> </span><span class="kw">sum</span>(bootstrapped_d <span class="op">&gt;</span><span class="st"> </span>fox_d) <span class="op">/</span><span class="st"> </span>B
<span class="kw">hist</span>(bootstrapped_d, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">10000</span>, <span class="kw">max</span>(<span class="kw">max</span>(bootstrapped_d), fox_d))<span class="op">+</span><span class="dv">20</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>fox_d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-20-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-20-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(fox_p)</code></pre></div>
<pre><code>[1] 4e-04</code></pre>
</div>
</div>
<div id="targeted-analysis-of-transcription-factors" class="section level1">
<h1>4. Targeted analysis of transcription factors</h1>
<div id="a-all-tfs" class="section level2">
<h2>(a) all TFs</h2>
<p>We found the external data base of TF2DNA that shows the target genes for each transcription factor. Since it offers a list of transcription factors as well as their target genes for 1306 transcription factors, we focus only on those transcription factors. For target genes, we use only those with the top binding scores.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(data.table)
TFs =<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;/Volumes/nicolae-lab/users/tae/TF_targets/Homo-sapiens_theoretical_TF2DNA/&quot;</span>)
TFs =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(TFs, <span class="st">&quot;.pscan&quot;</span>), <span class="cf">function</span>(x) x[[<span class="dv">1</span>]])</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results =<span class="st"> </span>ref[<span class="kw">match</span>(TFs, ref<span class="op">$</span>hgnc_symbol), ]
results =<span class="st"> </span>results[<span class="op">!</span><span class="kw">is.na</span>(results<span class="op">$</span>hgnc_symbol), ]
out =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">TF =</span> results<span class="op">$</span>hgnc_symbol, 
                 <span class="dt">numTargets =</span> <span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(results)),
                 <span class="dt">degree =</span> <span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(results)))</code></pre></div>
<p>For each transcription factor, we need to read in the target data, which can take very long due to a high number of file IO.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(results<span class="op">$</span>hgnc_symbol)){
  targets =<span class="st"> </span><span class="kw">fread</span>(<span class="kw">paste0</span>(<span class="st">&quot;/Volumes/nicolae-lab/users/tae/TF_targets/Homo-sapiens_theoretical_TF2DNA/&quot;</span>, results<span class="op">$</span>hgnc_symbol[i], <span class="st">&quot;.pscan&quot;</span>))
  targets =<span class="st"> </span>targets[<span class="kw">which</span>(targets<span class="op">$</span>p_value <span class="op">==</span><span class="st"> </span><span class="kw">min</span>(targets<span class="op">$</span>p_value)), <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">6</span>,<span class="dv">7</span>)]
  target_genenames =<span class="st"> </span>ref[<span class="kw">match</span>(targets<span class="op">$</span>target_name, ref<span class="op">$</span>hgnc_symbol), ]
  main_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">==</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id[i])
  target_ind =<span class="st"> </span><span class="kw">match</span>(target_genenames<span class="op">$</span>ensembl_gene_id, <span class="kw">colnames</span>(Y))
  target_ind =<span class="st"> </span>target_ind[<span class="op">!</span><span class="kw">is.na</span>(target_ind)]
  W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,main_ind], Y[, target_ind])
  scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W)
  out<span class="op">$</span>numTargets[i] =<span class="st"> </span><span class="kw">ncol</span>(W)
  out<span class="op">$</span>degree[i] =<span class="st"> </span><span class="kw">sum</span>(scores)
}
<span class="kw">head</span>(out)</code></pre></div>
<pre><code>     TF numTargets    degree
1  ADNP        647 509.58348
2 ADNP2        514 424.00513
3 AEBP2        140  74.75281
4  ALX1        316 205.13910
5  ALX3        465 308.13360
6  ALX4        316 427.57626</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d =<span class="st"> </span>out<span class="op">$</span>degree <span class="op">/</span><span class="st"> </span>out<span class="op">$</span>numTargets
df =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">d =</span> d, 
                <span class="dt">genes =</span> results<span class="op">$</span>hgnc_symbol, 
                <span class="dt">xaxis =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(d))
<span class="kw">plot</span>(d,
     <span class="dt">main =</span> <span class="st">&#39;Targeted TF&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;averaged scores&#39;</span>,
     <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(d) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>),
     <span class="dt">xlab =</span> <span class="kw">length</span>(d) <span class="op">+</span><span class="st"> </span><span class="dv">25</span>)
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">3</span>, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)
ind =<span class="st"> </span><span class="kw">which</span>(d <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>)
df =<span class="st"> </span>df[ind,]
<span class="kw">with</span>(df, <span class="kw">text</span>(d<span class="op">~</span>xaxis, <span class="dt">labels=</span>genes, <span class="dt">pos=</span><span class="dv">3</span>))</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-24-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-24-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="b-foxm1-analysis-with-target-genes" class="section level2">
<h2>(b) FOXM1 analysis with target genes</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">i =<span class="st"> </span><span class="kw">which</span>(results<span class="op">$</span>hgnc_symbol<span class="op">==</span><span class="st">&quot;FOXM1&quot;</span>)
targets =<span class="st"> </span><span class="kw">fread</span>(<span class="kw">paste0</span>(<span class="st">&quot;/Volumes/nicolae-lab/users/tae/TF_targets/Homo-sapiens_theoretical_TF2DNA/&quot;</span>, results<span class="op">$</span>hgnc_symbol[i], <span class="st">&quot;.pscan&quot;</span>))
targets =<span class="st"> </span>targets[<span class="kw">which</span>(targets<span class="op">$</span>p_value <span class="op">==</span><span class="st"> </span><span class="kw">min</span>(targets<span class="op">$</span>p_value)), <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">6</span>,<span class="dv">7</span>)]
target_genenames =<span class="st"> </span>ref[<span class="kw">match</span>(targets<span class="op">$</span>target_name, ref<span class="op">$</span>hgnc_symbol), ]
main_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">==</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id[i])
target_ind =<span class="st"> </span><span class="kw">match</span>(target_genenames<span class="op">$</span>ensembl_gene_id, <span class="kw">colnames</span>(Y))
target_ind =<span class="st"> </span>target_ind[<span class="op">!</span><span class="kw">is.na</span>(target_ind)]
W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,main_ind], Y[, target_ind])
scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W)
diff_cor_fox =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> <span class="kw">colnames</span>(orig)[target_ind],
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,fox_ind], 
                                          (orig[AA_ind,target_ind]))),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[<span class="op">-</span>AA_ind,fox_ind], 
                                          (orig[<span class="op">-</span>AA_ind,target_ind]))))

<span class="kw">plot</span>(diff_cor_fox<span class="op">$</span>AA <span class="op">~</span><span class="st"> </span>diff_cor_fox<span class="op">$</span>EA, <span class="dt">cex=</span><span class="fl">0.3</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">main=</span><span class="st">&#39;FOXM1&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;African Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;European Americans&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-25-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-25-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="new-permutation-test-for-foxm1" class="section level2">
<h2>new permutation test for FOXM1</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d =<span class="st"> </span><span class="kw">get_degree_c</span>(X<span class="op">$</span>A, Y[,fox_ind], Y[,target_ind])
out =<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">bootstrap_c</span>(X<span class="op">$</span>A, B, W))
p =<span class="st"> </span><span class="kw">sum</span>(out<span class="op">&gt;</span>d)<span class="op">/</span>B
<span class="kw">print</span>(p)</code></pre></div>
<pre><code>[1] 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(out, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(d<span class="op">+</span><span class="dv">10</span>, <span class="kw">max</span>(out))), <span class="dt">main =</span> <span class="st">&quot;FOXM1 permutation test&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-26-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-26-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="new-gene-zscan3" class="section level2">
<h2>new gene ZSCAN3</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">i =<span class="st"> </span><span class="kw">which</span>(results<span class="op">$</span>hgnc_symbol<span class="op">==</span><span class="st">&quot;ZSCAN30&quot;</span>)
targets =<span class="st"> </span><span class="kw">fread</span>(<span class="kw">paste0</span>(<span class="st">&quot;/Volumes/nicolae-lab/users/tae/TF_targets/Homo-sapiens_theoretical_TF2DNA/&quot;</span>, results<span class="op">$</span>hgnc_symbol[i], <span class="st">&quot;.pscan&quot;</span>))
targets =<span class="st"> </span>targets[<span class="kw">which</span>(targets<span class="op">$</span>p_value <span class="op">==</span><span class="st"> </span><span class="kw">min</span>(targets<span class="op">$</span>p_value)), <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">6</span>,<span class="dv">7</span>)]
target_genenames =<span class="st"> </span>ref[<span class="kw">match</span>(targets<span class="op">$</span>target_name, ref<span class="op">$</span>hgnc_symbol), ]
main_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">==</span><span class="st"> </span>results<span class="op">$</span>ensembl_gene_id[i])
target_ind =<span class="st"> </span><span class="kw">match</span>(target_genenames<span class="op">$</span>ensembl_gene_id, <span class="kw">colnames</span>(Y))
target_ind =<span class="st"> </span>target_ind[<span class="op">!</span><span class="kw">is.na</span>(target_ind)]
W =<span class="st"> </span><span class="kw">store_W_c</span>(Y[,main_ind], Y[, target_ind])
scores =<span class="st"> </span><span class="kw">get_score_W_c</span>(X<span class="op">$</span>A, W)
diff_cor_fox =<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">genes =</span> <span class="kw">colnames</span>(orig)[target_ind],
                      <span class="dt">AA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[AA_ind,fox_ind], 
                                          (orig[AA_ind,target_ind]))),
                      <span class="dt">EA =</span> <span class="kw">as.numeric</span>(<span class="kw">cor</span>(orig[<span class="op">-</span>AA_ind,fox_ind], 
                                          (orig[<span class="op">-</span>AA_ind,target_ind]))))

<span class="kw">plot</span>(diff_cor_fox<span class="op">$</span>AA <span class="op">~</span><span class="st"> </span>diff_cor_fox<span class="op">$</span>EA, <span class="dt">cex=</span><span class="fl">0.3</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="dt">main=</span><span class="st">&#39;ZSCAN30&#39;</span>,
     <span class="dt">ylab =</span> <span class="st">&#39;African Americans&#39;</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;European Americans&#39;</span>); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-27-1.png" width="500" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-27-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">zscan_ind =<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(Y) <span class="op">%in%</span><span class="st"> </span>results[results<span class="op">$</span>hgnc_symbol<span class="op">==</span><span class="st">&quot;ZSCAN30&quot;</span>, <span class="dv">1</span>])
d =<span class="st"> </span><span class="kw">get_degree_c</span>(X<span class="op">$</span>A, Y[,fox_ind], Y[,target_ind])
out =<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">bootstrap_c</span>(X<span class="op">$</span>A, B, W))
p =<span class="st"> </span><span class="kw">sum</span>(out<span class="op">&gt;</span>d) <span class="op">/</span><span class="st"> </span>B
<span class="kw">print</span>(p)</code></pre></div>
<pre><code>[1] 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(out, <span class="dv">50</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(d<span class="op">+</span><span class="dv">10</span>, <span class="kw">max</span>(out))), <span class="dt">main =</span> <span class="st">&quot;ZSCAN30 permutation test&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>d, <span class="dt">col =</span> <span class="st">&#39;red&#39;</span>)</code></pre></div>
<p><img src="figure/AA_EA.Rmd/unnamed-chunk-28-1.png" width="700" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tk382/networkAnc/blob/432088c570ebbca5ba2f586d1a018ddf2123049c/docs/figure/AA_EA.Rmd/unnamed-chunk-28-1.png" target="_blank">432088c</a>
</td>
<td>
tk382
</td>
<td>
2019-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] biomaRt_2.36.1    KEGGREST_1.22.0   ggplot2_3.1.0     nleqslv_3.3.2    
[5] data.table_1.12.0 mvtnorm_1.0-10    MASS_7.3-51.3    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1                prettyunits_1.0.2        
 [3] png_0.1-7                 Biostrings_2.50.2        
 [5] assertthat_0.2.1          rprojroot_1.3-2          
 [7] digest_0.6.18             R6_2.4.0                 
 [9] plyr_1.8.4                backports_1.1.3          
[11] stats4_3.5.3              RSQLite_2.1.1            
[13] evaluate_0.13             httr_1.4.0               
[15] pillar_1.3.1              zlibbioc_1.26.0          
[17] rlang_0.3.3               progress_1.2.0           
[19] curl_3.3                  lazyeval_0.2.2           
[21] whisker_0.3-2             blob_1.1.1               
[23] R.oo_1.22.0               S4Vectors_0.20.1         
[25] R.utils_2.8.0             rmarkdown_1.12           
[27] stringr_1.4.0             RCurl_1.95-4.12          
[29] bit_1.1-14                munsell_0.5.0            
[31] compiler_3.5.3            xfun_0.5                 
[33] pkgconfig_2.0.2           BiocGenerics_0.26.0      
[35] htmltools_0.3.6           tidyselect_0.2.5         
[37] tibble_2.1.1              workflowr_1.2.0          
[39] IRanges_2.16.0            XML_3.98-1.19            
[41] crayon_1.3.4              dplyr_0.8.0.1            
[43] withr_2.1.2               bitops_1.0-6             
[45] R.methodsS3_1.7.1         grid_3.5.3               
[47] gtable_0.3.0              DBI_1.0.0                
[49] git2r_0.25.2              magrittr_1.5             
[51] scales_1.0.0              stringi_1.4.3            
[53] XVector_0.20.0            RcppArmadillo_0.8.600.0.0
[55] fs_1.2.7                  tools_3.5.3              
[57] bit64_0.9-7               Biobase_2.40.0           
[59] glue_1.3.1                purrr_0.3.2              
[61] hms_0.4.2                 parallel_3.5.3           
[63] yaml_2.2.0                AnnotationDbi_1.42.1     
[65] colorspace_1.4-1          memoise_1.1.0            
[67] knitr_1.22               </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
