---
title: "Analysis of African American Co-expression Network"
author: "Tae Kim"
date: "7/9/2019"
output:
  workflowr::wflow_html:
    toc: false
---

```{r setup, include=FALSE}
set.seed(20190707)
knitr::opts_chunk$set(echo = TRUE,
                      dpi = 100)
library(MASS)
library(mvtnorm)
library(data.table)
library(nleqslv)
library(ggplot2)
library(KEGGREST)
library(biomaRt)
source("R/utilities_wv.R")
Rcpp::sourceCpp("src/utilities.cpp")
hgnc = setDF(fread("data/not_build/hgnc_ensembl.txt"))
hgnc = hgnc[hgnc$`Ensembl gene ID` != "", ]
```


# 1. Data pre-processing

We use the RNA-seq expression level data provided by GTEx, which has been already pre-processed. We choose to study not-sun-exposed suprapubic skin tissue as a pilot study because we expect to see some ancestry effect on the expression level on the skin tissue. Genes were selected based on expression threshold of $>0.1$ RPKM in at least 10 individuals and $\geq$ reads in at least 10 individuals. Then the expression values were quantile normalized to the average empirical distribution observed across samples. That means, each gene has the same distribution across the individuals. Then they were inverse quantile normalized to fit standard normal distribution.

## (a) Load and clean the data

```{r dataclean}
load("data/skin_not_sun_exposed_suprapubic.Rdata")
exp = skin; rm(skin)
colnames(exp) = gsub("[.]", "-", colnames(exp))
ind = which(colSums(!is.na(exp)) == 0)
exp = exp[,-ind]
ind= which(rowSums(!is.na(exp)) == 0)
exp = exp[-ind, ]
invisible(gc())
finalA = read.table("data/finalA.txt", header=TRUE)
AAs = intersect(colnames(exp), finalA$subject)
EAs = colnames(exp)[! colnames(exp) %in% finalA$subject]
newA= data.frame(subject = AAs, A = finalA[match(AAs, finalA$subject), 2])
Y = exp[, c(AAs, EAs)]
rm(exp); invisible(gc())
EA_A = data.frame(subject = EAs, A = 0)
X = rbind(newA, EA_A)
rm(newA, EA_A, AAs, EAs, ind, finalA)
rownames(Y) = sapply(strsplit(rownames(Y), '[.]'), function(x) x[1])
Y = t(Y)
ind = which(X$A > 0)
Y = Y[ind,]
X = X[ind, ]
orig = Y
```

## (b) scale the data and regress out the mean

Wefirst scale the gene expression level matrix to have variance 1. Then we regress out ancestry from the gene expression level to remove the mean effect. 

```{r scale}
orig_A = X$A
X$A = scale(X$A) * sqrt(length(X$A)) / sqrt(length(X$A)-1)
Y = resid(lm(Y~X$A))
Y = scale(Y) * sqrt(length(X$A)) / sqrt(length(X$A)-1)
```

```{r refchisq}
#reference chi squared distribution to compare the null distribution
ref_chisq = qchisq(seq(0,1, length=ncol(Y)-1), 1)
```

```{r B}
# number of permutations
B = 5000
```

## (c) match ENSEMBL ID and hgnc symbols

Using data set from https://www.genenames.org/download/custom/, we match the Ensembl gene IDs to HGNC symbols for better readability.

```{r hgnctoensembl}
hgnc_names = hgnc[match(colnames(Y), hgnc$`Ensembl gene ID`), "Approved symbol"]
ref = data.frame(ensembl_gene_id = colnames(Y), hgnc_symbol = hgnc_names)
ref$ensembl_gene_id = as.character(ref$ensembl_gene_id)
ref$hgnc_symbol = as.character(ref$hgnc_symbol)
```

# 2. Transcription Factors against all other genes

Since transcription factors usually regulate the transcription of multiple genes, we believe they were appropriate targets to study their variance in activity across different ancestry. We use the list of transcription factors from http://www.tfcheckpoint.org/.

## (a) Read in TF data

```{r readTF, echo = TRUE}
TF = fread("data/not_build/TFcheckpoint.txt", header=TRUE)
TF = setDF(TF)
TF = TF[TF$TFClass_human == "TFclass", ]
TF = TF[,1]
TF_results = ref[match(TF, ref$hgnc), ]
genes = colnames(Y)[which(colnames(Y) %in% TF_results$ensembl_gene_id)]
numTFs = length(genes)
TF_results = TF_results[match(genes, TF_results$ensembl_gene_id), ]
```

## (b) Test all TFs against the rest of the genes

For each transcription factor, we calculate its score statistics against all other genes. Then we add up the scores for each transcription factor and see if any stands out.

```{r get_scores_allTFs}
scores = matrix(0, length(genes), ncol(Y)-1)
for (i in 1:length(genes)){
  ind = which(colnames(Y) == TF_results$ensembl_gene_id[i])
  W = store_W_c(Y[,ind], Y[, -ind])
  V = store_V_c(Y[,ind], Y[, -ind])
  scores[i,] = get_score_W_c(X$A, W, V)
}
```

## (c) plot the result

The plot below shows that some transcription factors stand out from the rest with high score statistics. The red line shows the expected value based on $\chi^2$ statistics. The right plot shows the qq-plot of score statistics of the strongest signal "FOXD1" against the expected $\chi_1^2$ distribution. Note that all the score statistics are highly correlated to one another, so that the calibration is poor. 

```{r plot_allTFs}
par(mfrow = c(1,2))
d = rowSums(scores)
df = data.frame(d = d, genes = TF_results$hgnc_symbol, xaxis = 1:length(d))
plot(d, 
     main = 'Transcription Factors',
     xlim = c(0, length(d)+100),
     ylim = c(0, max(d) + 10000),
     xlab = "",
     ylab = "d test statistic")
abline(h=25190, col='red')
ind = which(d > ncol(Y) + 180 * sqrt(2*ncol(Y)))
df = df[ind, ]
with(df, text(d ~ xaxis, labels = genes, pos = 3))

ref_chisq = qchisq(seq(0,1, length=ncol(Y)-1), 1)
for (i in 1:length(genes)){
    if(TF_results$hgnc_symbol[i] %in% c("FOXD1")){
      qqplot(ref_chisq, scores[i, ], 
            main = paste(TF_results$hgnc_symbol[i], 'unadjusted'),
            cex = 0.5,
            ylab = "observed distribution of score",
            xlab = "expected distribution")
      abline(0,1,col = 'red')
  }
}
```

\pagebreak

# 3. Investigating the top signals

```{r permtest_func, echo = FALSE}
make_permtest = function(gene_name, Y, A){
  results = ref[match(gene_name, ref$hgnc), ]
  ind = which(colnames(Y) %in% results$ensembl_gene_id)
  W = store_W_c(Y[,ind], Y[, -ind])
  V = store_V_c(Y[,ind], Y[, -ind])
  scores = get_score_W_c(A, W, V)
  d = get_degree_c(A, Y[,ind], Y[,-ind])
  out = bootstrap_c(A, B, W, V)
  bootstrapped_d = rowSums(out)
  p = sum(bootstrapped_d > d) / B
  return(list(ind = ind, scores = scores, d = d, bootstrap = out, bootstrapped_d = bootstrapped_d, p = p))
}

diagnosis_plot = function(permtest, genename, orig, Y, A){
  max_scores = sort(permtest$scores, decreasing=TRUE, index.return=TRUE)
  results = ref[match(colnames(Y[,-permtest$ind])[max_scores$ix[1:5]], ref$ensembl_gene_id), ]
  top_genes = data.frame(gene_name = results$hgnc_symbol, score = max_scores$x[1:5],
                ensembl = results$ensembl_gene_id)
  gene1 = top_genes$gene_name[1]
  gene2 = top_genes$gene_name[2]
  par(mfrow = c(1,2))
  plot(orig[, permtest$ind] * orig[, -permtest$ind][, max_scores$ix[1]] ~ A,
     ylab = paste0(genename," * ",gene1), xlab='Ancestry',
     main = paste0("top signals of ", genename))
  plot(orig[, permtest$ind] * orig[, -permtest$ind][, max_scores$ix[2]] ~ A,
     ylab = paste0(genename," * ",gene2), xlab='Ancestry')
}
```

We take the top 3 transcription factors and perform permutation test to make inference about their degree statistics. 

## (a) FOXD1 permutation test & top signals

In order to verify if FOXD1 truly has a signal, we perform permutation test. Note that the highest test statistic does not mean the lowest p-value. Since the null distribution of high-dimensional case is not defined in closed form, we need to perform the permutation test to make inference. The p-value is indeed very small. We also look at top 2 target genes whose co-expression with FOXD1 varies strongly with respect to ancestry. We plot the product of the expression levels of the pair, which is intended to show the estimate of correlation at a given ancestry level. For these selected gene pairs, they seem to be highly co-expressed for Africans and lowly co-expressed for Europeans.

```{r foxd_perm}
foxd1 = make_permtest("FOXD1", Y = Y, A = X$A)
par(mfrow = c(1,2))
hist(foxd1$bootstrapped_d, 50, xlim = c(10000, max(max(foxd1$bootstrapped_d), d))+20,
     main = paste0("p value=", foxd1$p), xlab = "Null distribution of d")
abline(v=foxd1$d, col = 'red')
par(mfrow = c(1,2))
diagnosis_plot(foxd1, "FOXD1", orig = orig, Y = Y, A = orig_A)
```

## (b) NCOR2 permutation test and top signals

We repeat the same analysis for the second strongest signal NCOR2.

```{r ncor_perm_first}
ncor = make_permtest("NCOR2", Y = Y, A = X$A)
par(mfrow = c(1,2))
hist(ncor$bootstrapped_d, 50, xlim = c(10000, max(max(ncor$bootstrapped_d), ncor$d))+20,
     main = paste0("p value=", ncor$p), xlab = "Null distribution of d")
abline(v = ncor$d, col = 'red')
par(mfrow = c(1,2))
diagnosis_plot(ncor, "NCOR2", orig = orig, Y = Y, A = orig_A)
```

## (c) ZEB2 permutation test and top signals

We repeat the same analysis for the second strongest signal ZEB2. 

```{r ncor_perm}
ncor = make_permtest("ZEB2", Y = Y, A = X$A)
par(mfrow = c(1,2))
hist(ncor$bootstrapped_d, 50, xlim = c(10000, max(max(ncor$bootstrapped_d), ncor$d))+20,
     main = paste0("p value=", ncor$p), xlab = "Null distribution of d")
abline(v = ncor$d, col = 'red')
par(mfrow = c(1,2))
diagnosis_plot(ncor, "ZEB2", orig = orig, Y = Y, A = orig_A)
```

# 4. Remove outliers (samples with ancestry less than 0.6)

We suspect that the outliers with low global ancestry could be driving most of the signals. So we repeat the same analysis after we remove the samples with global ancestry less than 0.6.

```{r remove_outliers}
plot(orig_A, ylab = 'global ancestry', xlab="", xaxt='n')
ind = which(orig_A <= 0.6)
points(ind, orig_A[ind], col = 'red')
remove = which(orig_A <= 0.6)
newY = orig[-remove,]
newX = X[-remove,]
newX$A = scale(newX$A) * sqrt(length(newX$A)) / sqrt(length(newX$A)-1)
newY = resid(lm(newY~newX$A))
newY = scale(newY) * sqrt(length(newX$A)) / sqrt(length(newX$A)-1)
```


## (a) Re-run the transcription factors

```{r tf_no_outliers}
scores = matrix(0, length(genes), ncol(newY)-1)
for (i in 1:length(genes)){
  ind = which(colnames(newY) == TF_results$ensembl_gene_id[i])
  W = store_W_c(newY[,ind], newY[, -ind])
  V = store_V_c(newY[,ind], newY[, -ind])
  scores[i,] = get_score_W_c(newX$A, W, V)
}

par(mfrow = c(1,2))
d = rowSums(scores)
df = data.frame(d = d, genes = TF_results$hgnc_symbol, xaxis = 1:length(d))
plot(d, 
     ylab = 'd', 
     main = 'Transcription Factors',
     xlim = c(0, length(d)+100),
     ylim = c(0, max(d) + 10000),
     cex = 0.5)
abline(h=25190, col='red')
ind = which(d > ncol(newY) + 200 * sqrt(2*ncol(newY)))
if(length(ind)>0){
  df = df[ind, ]
  with(df, text(d ~ xaxis, labels = genes, pos = 3))
}

ref_chisq = qchisq(seq(0,1, length=ncol(newY)-1), 1)
ind = which.max(d)
qqplot(ref_chisq, scores[ind,],
       main = TF_results$hgnc_symbol[ind],
       cex = 0.5)
abline(0,1,col = 'red')
```

## (b) Top signal: TWIST1

### (i) Permutation Test

```{r twist_perm}
twist = make_permtest("TWIST1", Y = newY, A = newX$A)
par(mfrow = c(1,2))
hist(twist$bootstrapped_d, 50, xlim = c(min(twist$bootstrapped_d)-100, max(max(twist$bootstrapped_d), twist$d)),
     main = paste0("p-value = ", twist$p))
abline(v = twist$d, col = 'red')
par(mfrow = c(1,2))
diagnosis_plot(twist, "TWIST1", orig = orig[-remove, ], Y = newY, A = newX$A)
```

# 5. Targeted analysis of transcription factors

## (a) Test all TFs against target genes

The database of TF2DNA that shows the target genes for each transcription factor. We study the 1306 transcription factors introduced in this data set. Instead of measuring the co-expression against all the other genes, we only look at teh score statistics against the target genes with top binding scores. 

```{r targeted_Tf}
library(data.table)
TFs = list.files("/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/")
TFs = sapply(strsplit(TFs, ".pscan"), function(x) x[[1]])
results = ref[match(TFs, ref$hgnc_symbol), ]
results = results[!is.na(results$hgnc_symbol), ]
out = data.frame(TF = results$hgnc_symbol, 
                 numTargets = rep(NA, nrow(results)),
                 degree = rep(NA, nrow(results)))
```

For each transcription factor, we need to read in the target data, which can take very long due to a high number of file IO. 

```{r targeted_TF_long}
for (i in 1:length(results$hgnc_symbol)){
  targets = fread(paste0("/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/", results$hgnc_symbol[i], ".pscan"))
  targets = targets[which(targets$p_value == min(targets$p_value)), c(1,2,6,7)]
  target_genenames = ref[match(targets$target_name, ref$hgnc_symbol), ]
  main_ind = which(colnames(Y) == results$ensembl_gene_id[i])
  target_ind = match(target_genenames$ensembl_gene_id, colnames(Y))
  target_ind = target_ind[!is.na(target_ind)]
  target_ind = target_ind[target_ind != main_ind]
  W = store_W_c(Y[,main_ind], Y[, target_ind])
  V = store_V_c(Y[,main_ind], Y[, target_ind])
  scores = get_score_W_c(X$A, W, V)
  out$numTargets[i] = ncol(W)
  out$degree[i] = sum(scores)
}
```

```{r plot_targeted_Tf}
d = out$degree / out$numTargets
df = data.frame(d = d, 
                genes = results$hgnc_symbol, 
                xaxis = 1:length(d))
plot(d,
     main = 'Targeted TF',
     ylab = 'averaged scores',
     ylim = c(0, max(d) + 0.7),
     xlab = length(d) + 25)
abline(h=1, col = 'red')
ind = which(d > 2.8)
df = df[ind,]
with(df, text(d~xaxis, labels=genes, pos=3))
```


## (b) Top Signals

### (i) BATF analysis with permutation test

```{r targetperm_function, echo = FALSE}
make_target_permtest = function(genename){
  i = which(results$hgnc_symbol==genename)
  targets = fread(paste0("/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/", results$hgnc_symbol[i], ".pscan"))
  targets = targets[which(targets$p_value == min(targets$p_value)), c(1,2,6,7)]
  target_genenames = ref[match(targets$target_name, ref$hgnc_symbol), ]
  main_ind = which(colnames(Y) == results$ensembl_gene_id[i])
  target_ind = match(target_genenames$ensembl_gene_id, colnames(Y))
  target_ind = target_ind[!is.na(target_ind)]
  W = store_W_c(Y[,main_ind], Y[, target_ind])
  V = store_V_c(Y[,main_ind], Y[, target_ind])
  scores = get_score_W_c(X$A, W, V)

  d = get_degree_c(X$A, Y[,main_ind], Y[,target_ind])
  out = rowSums(bootstrap_c(X$A, B, W, V))
  p = sum(out>d)/B
  return(list(d = d, p = p, out = out))
}
```


```{r batf, fig.width = 4, fig.height = 4}
batf = make_target_permtest("BATF")
hist(batf$out, 50, xlim = c(0, max(batf$d+10, max(batf$out))),
     main = paste0("p value = ", batf$p),
     xlab = "null distribution of d of BATF")
abline(v=batf$d, col = 'red')
```

### (ii) ZNF785 analysis with permutation test

```{r znf785, fig.width = 5, fig.height = 5}
znf = make_target_permtest("ZNF785")
hist(znf$out, 50, xlim = c(0, max(znf$d+10, max(znf$out))),
     main = paste0("p value = ", znf$p),
     xlab = "null distribution of d of ZNF785")
abline(v=znf$d, col = 'red')
```

### (iii) NCOR2 with targeted genes only

```{r ncor_targets, fig.width = 4, fig.height = 4}
ncor = make_target_permtest("NCOR2")
hist(ncor$out, 50, xlim = c(0, max(ncor$d+10, max(ncor$out))),
     main = paste0("p value = ", ncor$p),
     xlab = "null distribution of d of NCOR2")
abline(v=ncor$d, col = 'red')
```


We conducted different analyses with the same data sets, and found some overlapping signals that appeared across the different analyses such as NCOR2 and ZEB2. 
