---
title: "Comparison of Co-expression Network between African Americans and European Americans"
author: "Tae Kim"
date: "7/9/2019"
output:
  workflowr::wflow_html:
    toc: false
---

```{r setup, include=FALSE, fig.width = 5, fig.width = 5, fig.aligh="center"}
set.seed(20190406)
knitr::opts_chunk$set(echo = TRUE,
                      dpi = 100)
library(MASS)
library(mvtnorm)
library(data.table)
library(nleqslv)
library(ggplot2)
library(KEGGREST)
library(biomaRt)
source("R/utilities_wv.R")
Rcpp::sourceCpp("src/utilities.cpp")
hgnc = setDF(fread("data/not_build/hgnc_ensembl.txt"))
hgnc = hgnc[hgnc$`Ensembl gene ID` != "", ]
```


# 1. Data pre-processing

We use the RNA-seq expression level data provided by GTEx, which has been already pre-processed. We choose to study not-sun-exposed suprapubic skin tissue as a pilot study because we expect to see some ancestry effect on the expression level on the skin tissue. Genes were selected based on expression threshold of $>0.1$ RPKM in at least 10 individuals and $\geq$ reads in at least 10 individuals. Then the expression values were quantile normalized to the average empirical distribution observed across samples. That means, each gene has the same distribution across the individuals. Then they were inverse quantile normalized to fit standard normal distribution.

## (a) load and clean the data

We read the expression level data and match the individuals with the ancestry information.

```{r dataclean}
load("data/skin_not_sun_exposed_suprapubic.Rdata")
exp = skin; rm(skin)
colnames(exp) = gsub("[.]", "-", colnames(exp))
ind = which(colSums(!is.na(exp)) == 0)
exp = exp[,-ind]
ind= which(rowSums(!is.na(exp)) == 0)
exp = exp[-ind, ]
invisible(gc())
finalA = read.table("data/finalA.txt", header=TRUE)
AAs = intersect(colnames(exp), finalA$subject)
EAs = colnames(exp)[! colnames(exp) %in% finalA$subject]
newA= data.frame(subject = AAs, A = finalA[match(AAs, finalA$subject), 2])
Y = exp[, c(AAs, EAs)]
rm(exp); invisible(gc())
EA_A = data.frame(subject = EAs, A = 0)
X = rbind(newA, EA_A)
rm(newA, EA_A, AAs, EAs, ind, finalA)
rownames(Y) = sapply(strsplit(rownames(Y), '[.]'), function(x) x[1])
Y = t(Y)
orig = Y
dim(Y)
```

## (b) scale the data and regress out the mean

We code African Americans as 0 and European Americans as 1 to compare group by group coexpression difference. We also consider correlation matrix, so we scale the gene expression level matrix to have variance 1. Then we regress out ancestry from the gene expression level to remove the mean effect. 

```{r scale}
orig_A = X$A
X$A[X$A > 0] = 1
X$A = scale(X$A) * sqrt(length(X$A)) / sqrt(length(X$A)-1)
Y = resid(lm(Y~X$A))
Y = scale(Y) * sqrt(length(X$A)) / sqrt(length(X$A)-1)
```


```{r refchi}
ref_chisq = qchisq(seq(0,1, length=ncol(Y)-1), 1)
```

```{r B}
B = 5000
```

## (c) match ENSEMBL ID and hgnc symbols

https://www.genenames.org/download/custom/

```{r hgnc}
hgnc_names = hgnc[match(colnames(Y), hgnc$`Ensembl gene ID`), "Approved symbol"]
ref = data.frame(ensembl_gene_id = colnames(Y), hgnc_symbol = hgnc_names)
ref$ensembl_gene_id = as.character(ref$ensembl_gene_id)
ref$hgnc_symbol = as.character(ref$hgnc_symbol)
```


# 2. Transcription Factors against all other genes

Since transcription factors usually regulate the transcription of multiple genes, we believe they are appropriate targets to study their variance in activity across different ancestry. 

## (a) read TF data

```{r tf}
TF = fread("data/not_build/TFcheckpoint.txt", header=TRUE)
TF = setDF(TF)
TF = TF[TF$TFClass_human == "TFclass", ] #newly added!
TF = TF[,1]
results = ref[match(TF, ref$hgnc_symbol), ]
genes = colnames(Y)[which(colnames(Y) %in% results$ensembl_gene_id)]
results = results[match(genes, results$ensembl_gene_id), ]
```

## (b) test all the TFs

```{r tf2, fig.width=8, fig.height=4}
scores = matrix(0, length(genes), ncol(Y)-1)
for (i in 1:length(genes)){
  ind = which(colnames(Y) == results$ensembl_gene_id[i])
  W = store_W_c(Y[,ind], Y[, -ind])
  V = store_V_c(Y[,ind], Y[, -ind])
  scores[i,] = get_score_W_c(X$A, W, V)
}
```

```{r tf3}
par(mfrow = c(1,2))
d = rowSums(scores)
df = data.frame(d = d, genes = results$hgnc_symbol, xaxis = 1:length(d))
plot(d, 
     ylab = 'd', 
     main = 'Transcription Factors')
abline(h=25190, col='red')
ind = which(d > ncol(Y) + 150 * sqrt(2*ncol(Y)))
df = df[ind, ]
with(df, text(d ~ xaxis, labels = genes, pos = 4))

ref_chisq = qchisq(seq(0,1, length=ncol(Y)-1), 1)
for (i in 1:length(genes)){
    if(results$hgnc_symbol[i] %in% c("ZNF528")){
      qqplot(ref_chisq, scores[i, ], 
            main = paste(results$hgnc_symbol[i], 'unadjusted'),
            cex = 0.5)
      abline(0,1,col = 'red')
  }
}
```


# 3. Investigating the top signals: ZNF528

## (a) Permutation Test 
```{r}
gene = "ZNF528"
results = ref[match(gene, ref$hgnc_symbol), ]
znf_ind = which(colnames(Y) %in% results[results$hgnc_symbol=="ZNF528", 1])
W = store_W_c(Y[,znf_ind], Y[, -znf_ind])
V = store_V_c(Y[,znf_ind], Y[, -znf_ind])
znf_scores = get_score_W_c(X$A, W, V)
znf_d = get_degree_c(X$A, Y[,znf_ind], Y[,-znf_ind])
shuffled_A = shuffle_x_c(X$A, B)
out = bootstrap_c(X$A, B, W, V)
bootstrapped_d = rowSums(out)
znf_p = sum(bootstrapped_d > znf_d) / B
par(mfrow = c(1,2))
hist(bootstrapped_d, 50, xlim = c(10000, max(max(bootstrapped_d), znf_d))+100,
     main = paste0("p-value = ", znf_p))
abline(v=znf_d, col = 'red')

```

### (iii) Box plot

We inspect the raw data to see any patterns between Afircan Americans and European Americans.

```{r}
AA_ind = which(orig_A > 0)
EA_ind = which(orig_A <= 0)
znf_exp = as.matrix(orig[,znf_ind])
znf_exp_aa = znf_exp[AA_ind]
znf_exp_ea = znf_exp[EA_ind]

df = data.frame(exp = znf_exp, label = c(rep("African Americans", length(AA_ind)),rep("European Americans", length(EA_ind))))

boxplot(exp ~ label, data = df, lwd = 2, main = "distribution of expression of ZNF528")
stripchart(exp ~ label, vertical = TRUE, data = df, 
    method = "jitter", add = TRUE, pch = 4, col = 'blue', cex=0.5)
```

## (c) Correlation of ZNF528 against all other genes

```{r, fig.width=8, fig.height=4}
diff_cor_znf = data.frame(genes = colnames(orig)[-znf_ind],
                      AA = as.numeric(cor(orig[AA_ind,znf_ind], 
                                          (orig[AA_ind,-znf_ind]))),
                      EA = as.numeric(cor(orig[EA_ind,znf_ind], 
                                          (orig[EA_ind,-znf_ind]))))

par(mfrow = c(1,2))

plot(diff_cor_znf$AA ~ diff_cor_znf$EA, cex=0.3, ylim = c(-1,1), xlim = c(-1,1), main='ZNF528',
     ylab = 'African Americans',
     xlab = 'European Americans'); abline(0,1,col = 'red')
```


## (d) What happens when I reduce the sample size of European Americans?

```{r, fig.width=8, fig.height=4}
EA_subset = sample(EA_ind, size=31)

diff_cor_znf = data.frame(genes = colnames(orig)[-znf_ind],
                      AA = as.numeric(cor(orig[AA_ind,znf_ind], 
                                          (orig[AA_ind,-znf_ind]))),
                      EA = as.numeric(cor(orig[EA_subset,znf_ind], 
                                          (orig[EA_subset,-znf_ind]))))
par(mfrow = c(1,2))
plot(diff_cor_znf$AA ~ diff_cor_znf$EA, cex=0.3, ylim = c(-1,1), xlim = c(-1,1), main='ZNF528',
     ylab = 'African Americans',
     xlab = 'European Americans'); abline(0,1,col = 'red')
```


## (e) Repeat the permutation test with sampled Euopreans

```{r}
W = store_W_c(Y[c(AA_ind, EA_subset),znf_ind], Y[c(AA_ind, EA_subset), -znf_ind])
V = store_V_c(Y[c(AA_ind, EA_subset),znf_ind], Y[c(AA_ind, EA_subset), -znf_ind])
znf_scores = get_score_W_c(X$A[c(AA_ind, EA_subset)], W, V)
znf_d = get_degree_c(X$A[c(AA_ind, EA_subset)], Y[c(AA_ind, EA_subset),znf_ind], Y[c(AA_ind, EA_subset),-znf_ind])
out = bootstrap_c(X$A[c(AA_ind, EA_subset)], B, W, V)
bootstrapped_d = rowSums(out)
znf_p = sum(bootstrapped_d > znf_d) / B
hist(bootstrapped_d, 50, xlim = c(10000, max(max(bootstrapped_d), znf_d))+20,
     main = paste0("p-value = ", znf_p))
abline(v=znf_d, col = 'red')
```


# 4. Targeted analysis of transcription factors

## (a) all TFs

We found the external data base of TF2DNA that shows the target genes for each transcription factor. Since it offers a list of transcription factors as well as their target genes for 1306 transcription factors, we focus only on those transcription factors. For target genes, we use only those with the top binding scores.  

```{r}
library(data.table)
TFs = list.files("/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/")
TFs = sapply(strsplit(TFs, ".pscan"), function(x) x[[1]])
```

```{r}
results = ref[match(TFs, ref$hgnc_symbol), ]
results = results[!is.na(results$hgnc_symbol), ]
out = data.frame(TF = results$hgnc_symbol, 
                 numTargets = rep(NA, nrow(results)),
                 degree = rep(NA, nrow(results)))
```

For each transcription factor, we need to read in the target data, which can take very long due to a high number of file IO. 

```{r}
for (i in 1:length(results$hgnc_symbol)){
  targets = fread(paste0("/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/", results$hgnc_symbol[i], ".pscan"))
  targets = targets[which(targets$p_value == min(targets$p_value)), c(1,2,6,7)]
  target_genenames = ref[match(targets$target_name, ref$hgnc_symbol), ]
  main_ind = which(colnames(Y) == results$ensembl_gene_id[i])
  target_ind = match(target_genenames$ensembl_gene_id, colnames(Y))
  target_ind = target_ind[!is.na(target_ind)]
  target_ind = target_ind[target_ind != main_ind]
  W = store_W_c(Y[,main_ind], Y[, target_ind])
  V = store_V_c(Y[,main_ind], Y[, target_ind])
  scores = get_score_W_c(X$A, W, V)
  out$numTargets[i] = ncol(W)
  out$degree[i] = sum(scores)
}
```

```{r}
d = out$degree / out$numTargets
df = data.frame(d = d, 
                genes = results$hgnc_symbol, 
                xaxis = 1:length(d))
plot(d,
     main = 'Targeted TF',
     ylab = 'averaged scores',
     ylim = c(0, max(d) + 1),
     xlab = length(d) + 25)
abline(h=1, col = 'red')
ind = which(d > 2.5)
df = df[ind,]
with(df, text(d~xaxis, labels=genes, pos=3))
```

## (b) ZNF528 analysis with target genes

```{r, fig.width = 5, fig.height = 5}
i = which(results$hgnc_symbol=="ZNF528")
targets = fread(paste0("/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/", results$hgnc_symbol[i], ".pscan"))
targets = targets[which(targets$p_value == min(targets$p_value)), c(1,2,6,7)]
target_genenames = ref[match(targets$target_name, ref$hgnc_symbol), ]
main_ind = which(colnames(Y) == results$ensembl_gene_id[i])
target_ind = match(target_genenames$ensembl_gene_id, colnames(Y))
target_ind = target_ind[!is.na(target_ind)]
W = store_W_c(Y[,main_ind], Y[, target_ind])
V = store_V_c(Y[,main_ind], Y[, target_ind])
scores = get_score_W_c(X$A, W, V)
diff_cor_fox = data.frame(genes = colnames(orig)[target_ind],
                      AA = as.numeric(cor(orig[AA_ind,main_ind], 
                                          (orig[AA_ind,target_ind]))),
                      EA = as.numeric(cor(orig[-AA_ind,main_ind], 
                                          (orig[-AA_ind,target_ind]))))

plot(diff_cor_fox$AA ~ diff_cor_fox$EA, cex=0.3, ylim = c(-1,1), xlim = c(-1,1), main='ZNF528',
     ylab = 'African Americans',
     xlab = 'European Americans'); abline(0,1,col = 'red')
```

## new permutation test for ZNF528

```{r}
d = get_degree_c(X$A, Y[,main_ind], Y[,target_ind])
out = rowSums(bootstrap_c(X$A, B, W, V))
p = sum(out>d)/B
print(p)
hist(out, 50, xlim = c(0, max(d+10, max(out))), main = paste0("p-value = ", p))
abline(v=d, col = 'red')
```

```{r}
max_scores = sort(scores, decreasing=TRUE, index.return=TRUE)
tmpresults = ref[match(colnames(Y[,target_ind])[max_scores$ix[1:5]], ref$ensembl_gene_id), ]
top_genes = data.frame(gene_name = tmpresults$hgnc_symbol, score = max_scores$x[1:5],
              ensembl = tmpresults$ensembl_gene_id)
gene1 = top_genes$gene_name[1]
gene2 = top_genes$gene_name[2]
par(mfrow = c(1,2))
plot(orig[, main_ind] * orig[, target_ind][, max_scores$ix[1]] ~ orig_A,
   ylab = paste0("ZNF528"," * ",gene1), xlab='Ancestry',
   main = paste0("top signals of ", "ZNF528"))
plot(orig[, main_ind] * orig[, target_ind][, max_scores$ix[2]] ~ orig_A,
ylab = paste0("ZNF528"," * ",gene2), xlab='Ancestry')
```

## new gene RREB1

```{r, fig.width = 5, fig.height = 5}
i = which(results$hgnc_symbol=="RREB1")
targets = fread(paste0("/Volumes/tae/Work/network/diffNet/data/pscan_files/Homo-sapiens_theoretical_TF2DNA/", results$hgnc_symbol[i], ".pscan"))
targets = targets[which(targets$p_value == min(targets$p_value)), c(1,2,6,7)]
target_genenames = ref[match(targets$target_name, ref$hgnc_symbol), ]
main_ind = which(colnames(Y) == results$ensembl_gene_id[i])
target_ind = match(target_genenames$ensembl_gene_id, colnames(Y))
target_ind = target_ind[!is.na(target_ind)]
W = store_W_c(Y[,main_ind], Y[, target_ind])
V = store_V_c(Y[,main_ind], Y[, target_ind])
scores = get_score_W_c(X$A, W, V)
diff_cor_fox = data.frame(genes = colnames(orig)[target_ind],
                      AA = as.numeric(cor(orig[AA_ind,main_ind], 
                                          (orig[AA_ind,target_ind]))),
                      EA = as.numeric(cor(orig[-AA_ind,main_ind], 
                                          (orig[-AA_ind,target_ind]))))

plot(diff_cor_fox$AA ~ diff_cor_fox$EA, cex=0.3, ylim = c(-1,1), xlim = c(-1,1), main='RREB1',
     ylab = 'African Americans',
     xlab = 'European Americans'); abline(0,1,col = 'red')
```


```{r}
d = get_degree_c(X$A, Y[,main_ind], Y[,target_ind])
out = rowSums(bootstrap_c(X$A, B, W, V))
p = sum(out>d) / B
print(p)
hist(out, 50, xlim = c(0, max(d+10, max(out))), main = paste0("p-value = ", p))
abline(v=d, col = 'red')
```

```{r}
max_scores = sort(scores, decreasing=TRUE, index.return=TRUE)
results = ref[match(colnames(Y[,target_ind])[max_scores$ix[1:5]], ref$ensembl_gene_id), ]
top_genes = data.frame(gene_name = results$hgnc_symbol, score = max_scores$x[1:5],
              ensembl = results$ensembl_gene_id)
gene1 = top_genes$gene_name[1]
gene2 = top_genes$gene_name[2]
par(mfrow = c(1,2))
plot(orig[, main_ind] * orig[, target_ind][, max_scores$ix[1]] ~ orig_A,
   ylab = paste0("RREB1"," * ",gene1), xlab='Ancestry',
   main = paste0("top signals of ", "RREB1"))
plot(orig[, main_ind] * orig[, target_ind][, max_scores$ix[2]] ~ orig_A,
ylab = paste0("RREB1"," * ",gene2), xlab='Ancestry')
```


